\section{Simulation procedure}

\subsection{Simulation of protein expression dynamics}
\label{metabolism:methods:simulation}

Default parameter values were based on approximate transcript and protein synthesis and turnover rates for animal cells reported in the literature \cite{Milo2016}, while gene activation and decay rates were arbitrarily set to a significantly faster timescale. Default feedback strengths for repressors acting at the gene, transcript, or protein levels were chosen such that $\sim$25-50\% of simulations failed to reach the threshold under normal conditions when one of two identical repressors was lost. Population-wide expression dynamics were estimated by simulating 5000 output trajectories in response to a three-hour transient step input to the gene activation rate. Simulations were performed using a custom implementation of the stochastic simulation algorithm \cite{Gillespie1977} (see Appendix \ref{appendix:software}). The algorithm constrains solutions to the set of discrete positive values, consistent with linearization about a basal level of zero gene activity. This simplifying assumption is based on the near-zero basal activities expected in the experimental systems, but is not required to support the conclusions of the model (Figs. \ref{fig:metabolism:figS2a}D and \ref{fig:metabolism:figS4}D).

\subsection{Evaluation of error frequencies and changes in expression dynamics}
\label{metabolism:methods:overexpression}

Gene expression trajectories were simulated both with (full repression) and without (partial repression) a second repressor. The time point at which the full-repression simulations mean level reached 30\% of its maximum value was taken to be the commitment time. At this time, a threshold for developmental success was set at the 99\textsuperscript{th} percentile of protein levels subject to full-repression. Error frequencies were obtained by evaluating the fraction of simulated protein levels that exceeded this threshold. Per this definition, the minimum possible error frequency is one percent. For simplicity we subtracted this percentage point from all reported error frequencies.

Protein expression dynamics were compared by evaluating the fraction of partially-repressed simulation trajectories in excess of the 99\textsuperscript{th} percentile of fully-repressed trajectories at each point in time. These fractions were then averaged across the time course, beginning with the reception of the input signal and ending at the previously defined commitment time. Each fraction may be thought of as the instantaneous error frequency, and their average reflects the extent to which the expression dynamics differ between the two sets of simulated trajectories (Fig. \ref{fig:metabolism:figS3a}).

\begin{figure}[h!]
\centering
\includegraphics[scale=1.0]{./figure_S3a}
\caption[Evaluation of protein overexpression metric.]{\textbf{Graphical depiction of the method used to quantify the impact of repressor loss on protein expression dynamics.} (A) Confidence bands span the 1\textsuperscript{st} to 99\textsuperscript{th} quantiles of protein levels simulated under full repression (grey) and under partial repression (purple), where a repressor is lost. The dashed purple line denotes the lower bound of the purple confidence band. The symbol $\tau$ denotes the commitment time as defined previously. (B) Loss of a repressor causes protein overexpression $E(t)$, which is calculated as the fraction of simulations that exceed the confidence band observed under full repression (grey) at a given time point. Orange-brown color scale reflects the value of $E(t)$ for each time point in the time course. Percent overexpression is calculated as a percent of simulations that exceed the confidence band integrated over the entire time course. A maximum of 100\% overexpression would occur when all simulations exceed the confidence band at all timepoints.}
\label{fig:metabolism:figS3a}
\end{figure}

\section{Dependence of model parameters on metabolic conditions}
\label{metabolism:methods:conditions}

IPC ablation reduces cellular glucose consumption. Presumably this would affect either the production and consumption of ATP or the production and consumption of substrates for RNA and protein synthesis (or both). The precise effects are unknown, so we independently modeled each scenario. Since ATP concentration remains fairly constant when respiration is limited \cite{Brown1992}, ATP flux (and ATP synthesis) is assumed to decrease. Because transcription, translation, and protein degradation all require ATP turnover, we halved their rate parameters under conditions of reduced glucose consumption. Under conditions of reduced substrate availability for RNA/protein synthesis, we assumed that only transcription and translation rates are affected by limiting fluxes of nucleotides and amino acids. We assumed only the translation rate is affected under conditions of reduced ribosome number. These assumptions are encoded in the rate parameters as shown in Table \ref{metabolism:methods:conditions:rxns}.

% TABLE OF METABOLISM-DEPENDENCE OF LINEAR PROPENSITIES
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{table}[h!]
\centering
\footnotesize
\caption[Reaction rate parameter dependence on environmental conditions]{\textbf{Reaction rate parameter dependence on environmental conditions}}
\label{metabolism:methods:conditions:rxns}
\begin{tabular}{L{1in} C{0.75in} C{1in} C{1.35in} C{1.35in}}
\toprule
    & \multicolumn{4}{c}{\bfseries Condition}\\ \cmidrule(lr){2-5}
    \textbf{Reaction} & Normal & Reduced ATP & Reduced substrates & Reduced ribosomes \\ 
    \cmidrule(lr){1-5}
    Transcription & $k_2$ & $k_2/2$ & $k_2/2$ & $k_2$ \\
    Translation & $k_3$ & $k_3/2$ & $k_3/2$ & $k_3/2$ \\
    Protein decay & $\gamma_3$ & $\gamma_3/2$ & $\gamma_3$ & $\gamma_3$ \\
\bottomrule
\end{tabular}
\end{table}

In all cases, feedback strengths were reduced in order to account for the intermediate processes abstracted by each feedback element. Feedback strength parameters $\eta_i$ were reduced four-fold under conditions of reduced energy metabolism and reduced RNA/protein substrate availability. This scaling assumes that both transcription and translation occur within the arbitrarily complex regulatory motifs represented by each repressor. This is a reasonable assumption for repressor proteins such as transcription factors and kinases. For RNA repressors such as microRNAs, feedback strength parameters could instead be reduced only two-fold to account for their reduced transcription rates. However, microRNAs must be transcribed, processed, and act with effector proteins in order to repress their targets. These fourfold reductions in feedback strength correspond to fourfold reduction of the transcriptional feedback gain $K_{C1}$ and twofold reduction in the post-transcriptional and post-translational feedback gains $K_{C2}$ and $K_{C3}$. Feedback strength parameters $\eta_i$ were only reduced two-fold under reduced protein synthesis conditions. This implies that the transcriptional and post-transcriptional feedback gains $K_{C1}$ and $K_{C2}$ decrease twofold while the post-translational feedback gain $K_{C3}$ remains constant. Each of these dependencies are summarized in Table \ref{metabolism:methods:conditions:reg}. The corresponding dependencies for our two alternate models based upon nonlinear transcription kinetics were analogous to those used in the linear model, and are listed in Tables \ref{metabolism:methods:conditions:two} and \ref{metabolism:methods:conditions:hill}. The half-maximal occupancy level and Hill coefficients of transcriptional repressors were assumed to be independent of growth rate.

% TABLE OF METABOLISM-DEPENDENCE OF FEEDBACK PROPENSITIES
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\begin{table}[h!]
\centering
\footnotesize
\caption[Feedback strength dependence on environmental conditions]{\textbf{Feedback strength dependence on environmental conditions}}
\label{metabolism:methods:conditions:reg}
\begin{tabular}{L{1.5in} C{0.75in} C{1in} C{1.35in} C{1.35in}}
\toprule
    & \multicolumn{4}{c}{\bfseries Condition}\\ \cmidrule(lr){2-5}
    \textbf{Feedback strengths} & Normal & Reduced ATP & Reduced substrates & Reduced ribosomes \\ \cmidrule(lr){1-5}
    Transcriptional & $\eta_1$ & $\eta_1/4$ & $\eta_1/4$ & $\eta_1/2$ \\    
    Post-transcriptional & $\eta_2$ & $\eta_2/4$ & $\eta_2/4$ & $\eta_2/2$ \\
    Post-translational & $\eta_3$ & $\eta_3/4$ & $\eta_3/4$ & $\eta_3/2$ \\
\bottomrule
\end{tabular}
\end{table}

% TABLE OF METABOLISM-DEPENDENCE OF TWOSTATE PROPENSITIES
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\begin{table}[h!]
\centering
\footnotesize
\caption[Two-state model dependence on environmental conditions]{\textbf{Two-state model dependence on environmental conditions}}
\label{metabolism:methods:conditions:two}
\begin{tabular}{L{1.75in} C{0.75in} C{1in} C{1.5in}}
\toprule
	& \multicolumn{3}{c}{\bfseries Condition}\\ \cmidrule(lr){2-4}
    \textbf{Reaction} & Normal & Reduced ATP & Reduced ribosomes \\ 
	\midrule  
    Transcription & $k_R$ & $k_R/2$ & $k_R$ \\
    Translation & $k_P$ & $k_P/2$ & $k_P/2$ \\
    Protein decay & $\gamma_P$ & $\gamma_P/2$ & $\gamma_P$  \\ 
    Transcriptional feedback & $\eta_G$ & $\eta_G/4$ & $\eta_G/2$ \\
    Feedback on mRNA & $\eta_R$ & $\eta_R/4$ & $\eta_R/2$ \\
    Feedback on protein & $\eta_P$ & $\eta_P/4$ & $\eta_P/2$  \\
\bottomrule
\end{tabular}
\end{table}

% TABLE OF METABOLISM-DEPENDENCE OF HILL PROPENSITIES
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\begin{table}[h!]
\centering
\footnotesize
\caption[Hill kinetics model dependence on environmental conditions]{\textbf{Hill kinetics model dependence on environmental conditions}}
\label{metabolism:methods:conditions:hill}
%\begin{tabular}{l c c c}
\begin{tabular}{L{1.5in} C{1in} C{1in} C{1.5in}}
\toprule
    & \multicolumn{3}{c}{\bfseries Condition}\\ \cmidrule(lr){2-4}
    \textbf{Reaction} & Normal & Reduced ATP & Reduced ribosomes \\ \cmidrule(lr){1-4}
    Transcription & $k_R$ & $k_R/2$ & $k_R$ \\
    Translation & $k_P$ & $k_P/2$ & $k_P/2$ \\
    Protein decay & $\gamma_P$ & $\gamma_P/2$ & $\gamma_P$  \\
    Feedback on mRNA & $\eta_R$ & $\eta_R/4$ & $\eta_R/2$ \\
    Feedback on protein & $\eta_P$ & $\eta_P/4$ & $\eta_P/2$  \\
\bottomrule
\end{tabular}
\end{table}
