\graphicspath{ {./figures/ratio/} }

\section{\nameref*{ch:ratio}}
\label{appendix:methods:ratio}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\subsection{Genetics}
\label{appendix:methods:ratio:genetics}

The recombineered \textit{pnt-gfp} BAC transgene inserted into the VK00037 landing site was previously described in Boisclair-Lachance et al. (2014). Alleles $pnt^{\Delta 88}$ \cite{ONeill1994a} and $pnt^2$ (Bloomington Stock 2222) were used to render the endogenous \textit{pnt} gene null in the presence of \textit{pnt-gfp}. A single copy of \textit{pnt-gfp} rescued $pnt^{\Delta 88}/pnt^2$ to full viability and fertility (Fig. \ref{fig:ratio:figS1}A). Cell nuclei of developing eye-antennal discs were marked by recombining \textit{H2Av-mRFP} (Bloomington stock 23651) with $pnt^2$. Experiments measuring wild type dynamics of \textit{Pnt-GFP} were done by dissecting eye discs from white prepupae carrying $w^{1118}$ ; $pnt-gfp / pnt-gfp$ ; $pnt^{\Delta 88}/pnt^2$, $H2Av-mRFP$. \textit{Pnt} isoform-specific expression was detected using enhancer traps \textit{HS20} (gift from B. Shilo) and $pnt^{1277}$ (Bloomington stock 837), which report \textit{PntP1} and \textit{PntP2} transcription respectively by expressing LacZ \cite{Scholz1993}. \textit{pnt-gfp} and \textit{pnt} isoform-specific expression were compared in white prepupae carrying $w^{1118}$; $pnt-gfp/pnt-gfp$; $HS20/+$ and $w^{1118}$;$pnt-gfp/pnt-gfp$; $pnt^{1277}/pnt^{1277}$. \textit{Pnt} gene dosage experiments were done using $w^{1118}$; $pnt-gfp/+$ ; $pnt^{\Delta 88}/pnt^2$ (1x pnt) and $w^{1118}$; $pnt-gfp/pnt-gfp$ ; $pnt^{\Delta 88}/pnt^2$ (2x pnt). Notch activity was conditionally reduced using the $N^{ts1}$ temperature sensitive allele \cite{Shellenbarger1975}. $N^{ts1}/N^{ts1}$ ; $pnt-gfp/+$ animals were raised at the permissive temperature (18 \textdegree{}C) and shifted to the restrictive temperature (28.5 \textdegree{}C) as third instar larvae for 24 h. Animals exposed to the restrictive temperature that were transferred back to the permissive temperature had roughened eye phenotypes and a notched wing phenotype as adults, consistent with effective inhibition of Notch activity. Control larvae of the same genotype were grown at the permissive temperature until dissection. Both control and heat-treated larvae were sexed and only \textit{N} hemizygote males carrying $N^{ts1}/Y$ ; $pnt-gfp/+$ were dissected as white prepupae. EGFR activity was conditionally reduced by placing the null allele $egfr^{f24}$ - also known as $egfr^{CO}$ \cite{Clifford1989} \textit{in trans} to the thermo-sensitive allele $egfr^{tsla}$ \cite{Kumar1998}, as previously described \cite{Pelaez2015a}. The genotype was $w^{1118}$; $egfr^{tsla}$, $pnt-gfp/egfr^{24}$, $pnt-gfp$. Ras activation was achieved using a transgene expressing a $Ras1^{V12}$ mutant and driven by a 3x\textit{sev} enhancer and promoter \cite{Fortini1992} as previously described \cite{Pelaez2015a}. \textit{Pnt-gfp} in the Ras mutant background was measured using $w^{1118}$; $pnt-gfp$, $Sev>Ras^{v12}/pnt-gfp$;$pnt^2$,$H2Av-mRFP/+$. Controls animals carried $w^{1118}$; $pnt-gfp/pnt-gfp$, $pnt^2$, $H2Av-mRFP/+$. \textit{Yan} mutant eye clones were generated using the $yan^{833}$ null allele \cite{Webber2013}, \textit{ey>FLP} and the FRT40 crossover point. $Pnt^+$ tissue was labeled using the clonal marker $Ubi>mRFP_{nls}$ (Bloomington Stock 34500). Developing eyes were dissected from white prepupae carrying $w$, $ey>FLP$; $pnt-gfp$, $yan^{833}$, $FRT40A/pnt-gfp$, $Ubi>mRFP_{nls}$, $FRT40A$. Control discs to measure the GFP-mRFP fluorophore bleed-through were obtained from flies carrying $w$, $ey>FLP$; $pnt-gfp$, $Ubi>mRFP_{nls}$, $FRT40A/pnt-gfp$, $FRT40A$ or $w$, $ey>FLP$; $pnt-gfp$, $Ubi>mRFP_{nls}$, $FRT40A/CyO$.

\subsection{Immunohistochemistry}
\label{appendix:methods:ratio:immunohistochemistry}

Unless otherwise noted, Pnt-GFP and Yan were measured in developing animals raised at 21 \textdegree{} C, selected as white prepupae, and subsequently aged in humid chambers for 5-10 h. Eye-antennal discs were dissected in PBS, and fixed in 4 \% (w/v) paraformaldehyde/PBS for $\sim$45 min. Endogenous Yan protein was detected with the mouse monoclonal anti-Yan antibody 8B12 (Developmental Studies Hybridoma Bank, 1:200 dilution) and the secondary goat anti-mouse Pacific Blue antibody (Life Technologies, 1:200 dilution). Expression from the \textit{HS20} and $pnt^{1277}$ enhancer traps was detected using mouse anti-$\beta$-galactosidase 40-1a (Developmental Studies Hybridoma Bank, 1:50 dilution). H2Av-mRFP was used as a nuclear marker as previously described in Pel\'{a}ez et al. (2015). Discs were incubated in 1:1 (v/v) PBS:VectaShield (Vector Laboratories) for 45 min, then in 100\% VectaShield for an additional 45 min before mounting.

For experiments using \textit{yan} mutant clones, $N^{ts}$, or $EGFR^{ts}$ alleles, nuclei were stained with a 4',6-diamidino-2-phenylindole (DAPI) nuclear marker. Samples were fixed in 4\% paraformaldehyde, rinsed with PBS-Tween 0.5\%, and permeabilized with PBS-Triton X-100 0.1\% for 20 minutes at room temperature. Permeabilization was important to allow DAPI penetration without perturbing the fluorescence of the Pnt-GFP protein. After permeabilization, eye discs were incubated in a blocking solution containing PBS-Tween 0.1\% and 1\% normal goat serum for 30 minutes at room temperature. Primary and secondary antibodies were incubated each for 2 hours at room temperature. Antibodies used with DAPI were: mouse anti-Yan 8B12 (DHSB, 1/500) and goat anti-mouse Cy3 (1/2000, Jackson Immunoresearch). Discs were mounted in 0.5\% n-propyl-gallate, 0.1M Tris pH 8.0 and 90\% glycerol.

Samples were kept in the dark at -20 \textdegree{}C and imaged no later than 18-24 hr after fixation. In all cases, 1024 x 1024 16-bit images were captured using either a Zeiss LSM880 or a Leica SP5 confocal microscope equipped with 40X oil objectives. During imaging, discs were oriented with the equator parallel to the x-axis of the image. Optical slices were set at 0.8Âµm slices (45-60 optical slices) with an additional digital zoom of 1.2-1.4 to completely image eye discs from basal to apical surfaces. Images recorded a region of at least 6 rows of ommatidia on each side or the dorsal-ventral eye disc equator. All discs for a given condition were fixed, mounted, and imaged in parallel to reduce measurement error.

\subsection{Quantification of expression levels}
\label{appendix:methods:ratio:quantification}

Expression dynamics were inferred from confocal image stacks using an updated version of an existing segmentation and annotation pipeline \cite{Pelaez2015a}. The new pipeline includes \textit{FlyEye Silhouette}; an open-source package for macOS that integrates our image segmentation algorithm with a GUI for cell type annotation. Subsequent analysis and visualization procedures were implemented in Python.

In all cases, cell segmentation was performed using either H2Av-mRFP (Figs. \ref{fig:ratio:fig1}, \ref{fig:ratio:fig2}, \ref{fig:ratio:figS1}, \ref{fig:ratio:figS2}, \ref{fig:ratio:figS3}, and \ref{fig:ratio:figS6}) or DAPI (Figs. \ref{fig:ratio:fig4}, \ref{fig:ratio:fig5}, and \ref{fig:ratio:figS8}) signals as a reference channel for identification of cell nuclei boundaries. Each layer of the reference channel was segmented independently. A single contour containing each unique cell was manually selected and assigned a cell type using a custom graphic user interface. DAPI-stained discs were segmented using a separate script based on the watershed algorithm in order to mitigate the effect of bright spots caused by DAPI accumulation in nucleoli. Further care was taken to avoid annotating contours containing such nucleoli. For each annotated cell contour, expression measurements were obtained by normalizing the mean fluorescence of the Pnt-GFP and Yan antibody channels by the mean fluorescence of the reference channel. This normalization serves to mitigate variability due to potentially uneven sample illumination, segment area, and in the case of His-RFP, differences in protein expression capacity between cells.

\subsection{Conversion of distance to time}
\label{appendix:methods:ratio:distance_to_time}

Cell positions along the anterior-posterior axis were mapped to developmental time as described previously \cite{Pelaez2015a,Pelaez2016}. This is predicated on two assumptions: the furrow proceeds at a constant rate of one column of R8 neurons per two hours, and minimal cell migration occurs. We find no reason to discard these assumptions.

For each disc, Delaunay triangulations were used to estimate the median distance between adjacent columns of R8 neurons \cite{Fortune1992}. We used the median rather than the mean distance, as was used in our previous study, because it minimized the influence of non-adjacent R8s that were falsely identified by the triangulation. Dividing the furrow velocity of 2 h per column by this median distance yields a single conversion factor from position along the anterior-posterior axis to developmental time. This factor was applied to all cell measurements within the corresponding disc, yielding expression time series. Notably, these are not single cell dynamics, but rather aggregate dynamics across the development time course of a cell population.

\subsection{Computation of moving averages and confidence intervals}
\label{appendix:methods:ratio:moving_averages}

Moving averages were computed by first-order Savitzky-Golay filtration \cite{Savitzky1964}. This method augments the simple windowing approach used in our previous study \cite{Pelaez2015a} by enabling visualization of expression trends at early time-points that are otherwise obscured by large window sizes. A secondary first-order filtration with one-fifth the original window size was applied to smooth lines for visualization purposes.

None of our conclusions are sensitive to our choice of filtration or smoothing method \cite{Pelaez2015a}. Primary window sizes of 250 and 75 cells were used for reporting the expression of multipotent and differentiated cells, unless noted otherwise. Confidence intervals for the moving average were inferred from the 2.5th and 97.5th percentile of 1000 point estimates of the mean within each window. Point estimates were generated by bootstrap resampling (with replacement) the expression levels within each window.

\subsection{Alignment of expression data}
\label{appendix:methods:ratio:alignment}

Cells were aligned with a reference population by shifting them in time. The magnitude of this shift was determined by maximizing the cross-correlation of progenitor Pnt-GFP expression $Y(t)$ with the corresponding reference time series $X(t)$. Rather than raw measurements, moving averages within a window of ten cells were used to improve robustness against noise. This operation amounts to:
\begin{equation}
z = \argmax_{dt} \ {\hat{\gamma}_{X(t),Y(t)}}
\end{equation}
\begin{equation}
\hat{\gamma}_{X(t),Y(t)} (dt) = E \Big[ \frac{(Y(t+dt)-\mu_Y)(X(t+dt)-\mu_X)}{\sigma_Y \sigma_X} \Big]
\end{equation}
where, $\mu$ and $\sigma$ are the mean and standard deviation of each time series, and $dt$ is the time by which the population should be shifted.

For each experimental treatment, a disc was randomly chosen and shifted in time such that time zero corresponds to the first annotated R8 neuron. This disc then served as the reference population for the alignment of all subsequent biological replicates within the treatment. Similarly, different experimental treatments (e.g. control and perturbation) were aligned by first aligning the discs within each treatment, then aggregating all cells within each treatment and repeating the procedure with the first treatment serving as the reference.

This approach differs from the previous implementation of our pipeline in which discs were manually aligned by the inflection point of their Yan-YFP expression profiles \cite{Pelaez2015a}. Manual alignment entails arbitrary prioritization of certain dynamic features over others. Our revised protocol yields consistent, reproducible alignment of expression time series that equally weighs the entire time course. The automated approach is more principled but less robust than the manual approach. Specifically, it fails when dynamic forms qualitatively differ between experimental treatments. In these instances, we revert to manual alignment using the inflection point of Pnt-GFP induction as a reference.

\subsection{Simple competitive binding model}
\label{appendix:methods:ratio:simple_model}

Figure \ref{fig:ratio:figS4} presents results for an equilibrium model of two species, Yan ($Y$) and Pnt ($P$), competing for a finite pool of shared binding sites, $S$:
\begin{equation}
Y + S \xrightleftharpoons{\,K_{D,Yan}\,} SY
P + S \xrightleftharpoons{\,K_{D,Pnt}\,} SP
\end{equation}
where $K_{D,Yan}$ and $K_{D,Pnt}$ are equilibrium association constants and $SY$ and $SP$ denote the bound species. Applying a mass balance to the total protein and binding site ($S_0$) abundances:
\begin{equation}
Y_0 = Y + SY
P_0 = P + SP
S_0 = S + SY + SP
\end{equation}
yields an analytically tractable system of nonlinear equations \cite{Wang1995}. For each pair of absolute protein abundances $(Y_0,P_0)$ in Figure \ref{fig:ratio:figS4}B, the Pnt binding site occupancy is simply $SP/S_0$.

\subsection{Competitive binding model with cooperativity}
\label{appendix:methods:ratio:competitive_model}

The model presented in Figure \ref{fig:ratio:fig3} expands upon the work of Hope, Rebay, and Reinitz (2017). The model is based on a single \textit{cis}-regulatory element consisting of $n$ adjacent binding sites, each of which may be designated as ETS or non-ETS. Each binding site may only exist in one of three binding states; bound by a single copy of Yan, bound by a single copy of Pnt, or unbound. Thermodynamic potentials were assigned to each binding state using two parameters for each transcription factor. The parameter $\alpha_X$ defines the free energy of transcription factor $X$ binding to an ETS site, while $\beta_X$ defines the free energy of binding to a non-ETS site (Fig. \ref{fig:ratio:figS5}A). A unique configuration of binding states for all $n$ binding sites constitutes a single microstate, $k$. The thermodynamic potential of each microstate was taken to be the sum of thermodynamic potentials for each of its constituent binding sites. For each microstate, the stabilizing effect of polymerization was incorporated via a third parameter, $\gamma_X$, that defines the free energy of SAM-SAM binding between a pair of similar transcription factors bound to adjacent sites. The net result is a total thermodynamic potential, $\Delta G_k$, for each microstate. An example enumeration of all possible microstates for an element consisting of one ETS site preceding two non-ETS sites is provided in Figure \ref{fig:ratio:figS5}B. The statistical frequencies of each microstate were evaluated by constructing a canonical ensemble:
\begin{equation}
p_k = \frac{\displaystyle exp( \frac{-\Delta G_k}{RT} ) [P]^{a_P(k)}[Y]^{a_Y(k)} } {\displaystyle \sum_{k} {exp(\frac{-\Delta G_k}{RT})[P]^{a_P(k)}[Y]^{a_Y(k)}}}
\end{equation}
in which $p_k$ is the statistical frequency of microstate $k$, $[P]$ and $[Y]$ are the Pnt and Yan concentrations, $a_P(k)$ and $a_Y(k)$ are functions representing the number of bound molecules of $P$ and $Y$ within microstate $k$, $T$ is a fixed temperature set to 300 K, and $R$ is the gas constant. Fractional occupancies for each binding site correspond to the cumulative statistical frequency of all microstates in which the site is occupied by a given transcription factor. Overall fractional occupancies are similarly evaluated across all sites within the element.

We consider regulatory elements comprised of 12 binding sites in which only the first site carries the ETS designation. We retain the same parameterization of Yan binding proposed by Hope, Rebay, and Reinitz (2017): $\alpha_Y = -9.955 kcal mol^{-1}$, $\beta_Y = -5.837 kcal mol^{-1}$, and $\gamma_Y = -7.043 kcal mol^{-1}$. We parameterized Pnt binding thermodynamics to provide balanced competition between Pnt and Yan in the absence of any SAM-mediated polymerization of Pnt. That is, we set Pnt binding affinities such that the transition from Pnt to Yan occupancy occurs when Pnt and Yan concentrations are approximately equal. While a parameterization using experimentally measured data would improve predictive accuracy, our aim here is primarily to obtain insight. The model used to generate Fig. \ref{fig:ratio:fig3}D-F assumes that Pnt binds individual sites with elevated affinities $\alpha_P = 0.96 (\alpha_Y + \gamma_Y )$ and $\beta_P = 0.96 (\beta_Y + \gamma_Y )$. The model used to generate Fig. \ref{fig:ratio:fig3}A-C uses these same elevated binding affinities for Yan, while setting $\gamma_Y = 0 kcal mol^{-1}$. Qualitatively, our results are not sensitive to this parameterization. 

\subsection{Analysis of \textit{yan} clones}
\label{appendix:methods:ratio:clones}

We used \textit{ey>FLP} and \textit{FRT40A} to generate $yan^{833}$ null clones within 23 eye discs carrying the Pnt-GFP transgene (see Section \ref{ratio:genetics}). The chromosome carrying the wildtype \textit{yan} allele was marked with a Ubi-mRFPnls transgene, enabling automated detection of subpopulations with distinct \textit{yan} gene dosages, each characterized by a distinct level of mRFP fluorescence. Discs were dissected, fixed, and co-stained with DAPI prior to confocal imaging. Images of 36 unique vertical cross-sections spanning non-overlapping cells were collected in total. 

We deployed the methods developed in Chapter \ref{ch:clones} to measure the expression level of each reporter in each nucleus and automatically label each measurement as mutant, heterozygous, or homozygous for the Ubi-mRFPnls clonal marker (See Sections \ref{ch:clones:segmentation} and \ref{ch:clones:annotation}). For each segment, Ubi-mRFPnls and Pnt-GFP fluorescence was quantified by normalizing the average intensity of all pixels within the respective fluorescence channel by the average DAPI fluorescence. Segments containing less than 250 pixels were removed. 

Fluorescence bleedthrough between the RFP and GFP channels was visually apparent in these discs. To confirm our suspicion, control clones were generated in six wildtype \textit{yan} eye discs co-expressing Ubi-mRFPnls and Pnt-GFP. These are the same discs as those presented in Chapter \ref{ch:clones}. We used the correction strategy presented in Section \ref{ch:clones:correction} to systematically correct for bleedthrough from the Ubi-mRFPnls reporter into the GFP channel. The correction successfully eliminated any detectable difference in Pnt-GFP expression between Ubi-mRFPnls genotypes in the wildtype \textit{yan} control discs (not shown). The same procedure was therefore applied to all measurements of \textit{yan} null clones (Fig. \ref{fig:ratio:fig4}H).

We used the selection tool described in Section \ref{appendix:methods:clones:curation} to limit the analysis to cells taken from regions immediately posterior to the MF in each eye disc. We further limited the comparison to clonal genotypes that overlap in developmental time. These restrictions served to buffer against differences in developmental context and focus attention on the region of elevated Pnt-GFP expression. Using the approach described in Section \ref{appendix:methods:clones:comparison}, cells residing on the border of each clone were excluded from all comparisons to mitigate edge effects. The remaining measurements were aggregated across all eye discs for statistical comparison between clonal genotypes.

\subsection{Visualization of relative Pnt and Yan expression in \textit{Notch} mutant discs}
\label{appendix:methods:ratio:notch_images}

Visualizations were constructed by applying a smoothing operation to maximum intensity projections across confocal layers spanning progenitors (Fig. \ref{fig:ratio:figS5}A,B), and then mapping the absolute difference in Pnt-GFP and Yan antibody fluorescence to a diverging color scale. The smoothing operation consists of three sequential applications of a grey-closing filter followed by a single pass of a three-pixel wide median filter. This procedure dampens noise. Raw image fluorescence intensities were normalized to a 0-1 scale before application of any filters, so the maximum possible difference between Pnt-GFP and Yan fluorescence channels is unity. The color scale was truncated to a range of -0.3 to 0.3 for visualization purposes. No post-processing was applied to the maximum intensity projections in Figures \ref{fig:ratio:figS5}A and \ref{fig:ratio:figS5}B.

\subsection{Analysis of periodic spatial patterns in \textit{Notch} mutant discs}
\label{appendix:methods:ratio:autocorrelation}

Progenitor cells were selected from a 1.75 h window immediately posterior to the morphogenetic furrow. This window corresponds to approximately one column of eventual ommatidia. The window is identifiable in \textit{Notch} mutant discs because the MF serves as a reference. Digital spatial signals were assembled by sampling progenitor $log_2$-transformed Pnt to Yan ratios, $X$, as a function of cell position along the dorso-ventral axis, $y$. Both autocorrelation analysis and spectral decomposition were applied to these signals.

Autocorrelation functions were assembled by computing the moving average of expression similarity, $C$, as a function of dorso-ventral separation distance, $d$:
\begin{equation}
C_{ij} = \frac{ (X_i-E[X])(X_j-E[X]) }{ E[X^2] - E[X]^2 }
\end{equation}
\begin{equation}
d_{ij} = | y_i - y_j |
\end{equation}
where $E$ denotes the expected value, and $i$ and $j$ are indexed over all cells in order of increasing separation distance. Moving averages and confidence intervals were computed as described previously, with a window size of 50 sequential values.

Spatial signals were decomposed into spectral components via the Lomb-Scargle periodogram using the AstroML software package \cite{VanderPlas2012}. These periodograms were used rather than Fourier decomposition because they enable spectral decomposition of irregularly sampled signals \cite{VanderPlas2018}. Significance thresholds were inferred from the 95th percentile of peak spectral powers detected during repeated decomposition of 1000 null signals. Null signals were constructed by resampling signal intensities while maintaining constant sampling times.