



Cell fate decision mechanisms are notoriously difficult to quantitatively characterize. Chapter \ref{ch:clones} introduces a computational framework for automated quantitative analysis of genetic mosaics; a class of experiments designed to probe gene expression \textit{in vivo}. The framework combines computer vision and statistics to measure protein levels in individual cells and predict their respective genotypes. It thereby facilitates systematic quantitative comparison of expression levels between cells subject to control and perturbation conditions. The accompanying software eliminates each of the labor-intensive steps of a quantitative workflow, and will hopefully lower the barrier to adoption of quantitative analysis strategies. 

Chapter \ref{ch:ratio} explores a novel cell fate decision mechanism underlying photoreceptor specification in the developing fruit fly eye. Computer vision and statistical modeling techniques are used to extract quantitative measurements of transcription factor dynamics from a wealth of confocal microscopy data. These data show that differentiation is driven by dynamic changes in the ratio between two transcription factors, and is agnostic to changes in their absolute concentrations as long as the ratio remains constant. A general model based on the statistical physics of transcription factor DNA binding shows that this phenomenon is a natural consequence of competition between the two transcription factors for common binding sites. The findings add a new dimension to our understanding of how transcription factors coordinate cell fate decisions, and showcase the importance of both quantitative and dynamic measurements for characterizing developmental systems. 

Chapter \ref{ch:metabolism} proposes a novel theory to explain why the regulatory networks that coordinate cell fate decisions often contain several repressors tasked with attenuating expression of a single target gene. The theory posits that auxiliary negative regulators enable development to proceed more quickly by mitigating erroneous cell fate decisions when cells are rapidly metabolizing. It is supported by a robust collection of qualitative experiments showing that a broad variety of repressor loss-of-function phenotypes are reversed when biosynthesis rates are artificially slowed. A quantitative modeling framework is used survey a hypothesized mechanistic origin of this effect. Namely, that auxiliary repressors help avert erroneous decisions by expanding cells capacity to buffer excess protein expression. Quantitative measurements of transcription factor activity serve to validate the predictions made by the modeling framework. As shorter developmental times confer a selective advantage upon organisms, these findings may represent a novel evolutionary driving force for increased robustness of cell fate decisions.

Beyond their insights into the mechanics of cell fate decisions, these efforts have spawned several computational tools that may prove valuable to the broader community (see \ref{appendix:software}). All such resources have been made freely available, and their continued development should help contribute toward the sustained prevalence of quantitative analysis of cell fate decisions.
