\chapter{Conclusions}
\label{ch:conclusion}

This dissertation surveyed cell fate decisions through a quantitative lens. Its findings broadly provide a fresh perspective as to how the structure and function of gene regulatory networks control emergent behavior at the organismal scale. Each chapter advances the fields of systems and developmental biology in a unique way.

Chapter \ref{ch:clones} introduced a computational framework for automated quantitative analysis of genetic mosaics; a class of experiments designed to probe cell fate decisions \textit{in vivo}. The framework combines computer vision and statistics to measure protein levels in individual cells and infer their respective genotypes. It thereby enables automated and systematic comparison of cells subject to control and perturbation conditions in an otherwise equivalent background. The accompanying open-source software eliminates each of the labor-intensive steps of a quantitative workflow. Its release will make quantitative mosaic analysis more accessible to the broader research community, thus contributing toward the quantification of biology.

Chapter \ref{ch:ratio} explored a novel cell fate decision mechanism underlying photoreceptor specification in the larval eye. Computer vision techniques were used to extract quantitative measurements of transcription factor dynamics from a wealth of confocal microscopy data. Statistical analysis of these data revealed that differentiation is driven by dynamic changes in the ratio between two transcription factors, and is agnostic to changes in their absolute concentrations as long as the ratio remains constant. A general model based on the statistical physics of transcription factor DNA binding showed that this phenomenon is a natural consequence of competition between the two transcription factors for common binding sites. The findings add a new dimension to our understanding of how transcription factors coordinate cell fate decisions, and have broader implications for other developmental contexts in which multiple transcription factors control subsequent developmental events. They also rebuke the canonical model of photoreceptor specification in the larval eye \cite{Graham2010}, while rationalizing all existing experimental data \cite{Shwartz2013,BoisclairLachance2014,Pelaez2015a}. The results also exemplify the importance of both quantitative and dynamic measurements for characterizing developmental systems. 

Chapter \ref{ch:metabolism} proposed a new theory to explain why the regulatory networks that coordinate cell fate decisions often contain several repressors tasked with attenuating expression of a single target gene. The theory posits that auxiliary negative regulators enable development to proceed more quickly by mitigating erroneous cell fate decisions when cells are rapidly metabolizing. It is supported by a robust collection of qualitative experiments showing that a broad variety of repressor loss-of-function phenotypes are reversed when biosynthesis rates are artificially slowed. A quantitative modeling framework was used to explore the mechanistic origin of this effect. Namely, that auxiliary repressors help avert erroneous decisions by expanding cells capacity to buffer excess protein expression. Quantitative measurements of transcription factor activity served to validate the predictions made by the modeling framework. As shorter developmental times confer a selective advantage upon organisms, these findings may represent a novel evolutionary driving force for increased robustness of cell fate decisions.

=== EXPAND ===




\section{ A novel driving force in the evolution of GRN topologies }

- Cartoon analogy
- Relation to temperature - would require knowledge of temperature/metabolism scaling
- Introduction of simpler model
- Exploration of tradeoff between synthesis and degradation
- Potential for a complete theoretical framework
- Is redundant regulation the only link? What about promoters?





Each chapter of this dissertation leveraged the power of quantitative measurement and mathematical modeling to tease deeper meaning out of experimental data than would otherwise have been possible using conventional techniques.

ask how cells reliably commit to appropriate fates at the correct time and place.

tease explanations out of experimental data, rebuke existing dogma, propose new mechanisms, and postulate new theories. 

The findings of this dissertation provide a fresh perspective as to how the structure and function of gene regulatory networks control emergent behavior at the organismal scale.

They suggest 

These achievements were attained through principled application of existing knowledge from other technical disciplines.







\section{New computational resources for quantitative biologists}

The research presented in this dissertation spawned several computational tools that may prove valuable to the broader community of quantitative biologists. All of these resources have been made freely available online under open license for unrestricted use and future development. They are published alongside \textbf{FlyEye Silhouette}: \url{http://silhouette.amaral.northwestern.edu}, a GUI-based MacOS application for segmentation, quantification, and annotation of cell nuclei in the \textit{Drosophila} eye imaginal disc that was developed by Helio Tejedor in the Amaral lab at Northwestern University. The list below describes each of the new tools and their high level functions. All are accessible via GitHub repositories mirrored between both \href{https://github.com/sebastianbernasek/}{my personal account} and the \href{https://github.com/amarallab}{Amaral} and \href{https://github.com/bagherilab}{Bagheri} lab accounts. These repositories contain high level API documentation in addition to a series of Jupyter notebooks that walk the user through a series of usage examples. 

\begin{itemize}[leftmargin=*, topsep=10pt, parsep=5pt, partopsep=5pt, itemsep=10pt]
    
  % FLYEYE ANALYSIS
  \item \textbf{FlyEye Analysis}: \url{https://github.com/sebastianbernasek/flyeye}
  \newline
  Python framework for analyzing data generated using FlyEye Silhouette. Core features include inference of cell developmental ages and analysis of the resultant expression dynamics. Also provides tools to quantify expression heterogeneity and spatial patterns.
  
  % FLYEYE ANALYSIS
  \item \textbf{FlyEye Clones}: \url{https://github.com/sebastianbernasek/clones}  
  \newline
  Python framework for automated mosaic analysis of \textit{Drosophila} eye imaginal discs. Core features will be integrated with future versions of FlyEye Silhouette. 
  
  % FLYEYE SYCLONES
  \item \textbf{FlyEye SyClones}: \url{https://github.com/sebastianbernasek/syclones}
  \newline 
  Python framework for generating synthetic microscopy data that mimic key features of mosaic eye imaginal discs.
  
  % PolyTF BINDING
  \item \textbf{PolyTF Binding}: \url{https://github.com/sebastianbernasek/binding}
  \newline 
  Python framework for simulating the equilibrium occupancy of DNA binding sites by one or more polymerizing transcription factors. Utilizes a C backend that efficiently enumerates all possible microstates in a recursive fashion, enabling nested parallelization of the main computational bottleneck. For systems of two or more transcription factors, the implementation confers a major performance advantage over the sequential enumeration strategy proposed by the manuscript that inspired the model \cite{Hope2017}.
  
  % GENESSA
  \item \textbf{GeneSSA}: \url{https://github.com/sebastianbernasek/genessa}
  \newline
  Python framework for exact stochastic simulation of gene regulatory network dynamics \cite{Gillespie1977}. Simulations are executed by a C backend optimized for performance on networks with a narrow scope of pre-defined reaction propensity functions. The limited scope is by design; GeneSSA prioritizes computational efficiency at the expense of flexibility by explicitly hard coding a set of functional forms. This design places GeneSSA among the most performant implementations of the exact stochastic simulation algorithm for several common types of GRNs. The framework may be (and has been) extended to include additional kinetic formulations as they are required.
  
\end{itemize}

\textbf{NOTE}: See Appendix \ref{appendix:resources} for all experimental data and another set of Jupyter notebooks walk the user through reproducing all of the results and figures presented in this dissertation.











\section{ Tools for a quantitative future }

- Need for quantification in each set of experiments
- Danger of 'feedback' diagrams
- Need for quantitative measurements


\section{ Development viewed through the lens of control theory }

- Value of 'dynamics' only perspective
- Discussion on link to molecular mechanisms
- Ratio control circuit






% recent papers on metabolism & phenotypes --- see PNAS presentation and paper to rich/luis
% recent gap genes example, all information encoded in a handful of factors
% emphasis on dynamics - simple model, how does one process scale versus another



%\section{Dynamic perspective of cell behavior}
%	\subsection{Dynamics matter}
%		\paragraph{Decisions are localized in time.}
%	\subsection{Control theoretical insights into cell behavior}
%		\paragraph{Yeast papers}
%		\paragraph{Khammash papers}


\Chapter{Conclusions}

reconcile pnt/yan mechanism with yan over-expression

discuss impact of clones pipeline on pnt/yan study


- development of 'penetrance theory' based on analytical framework using moment equations
- rigorous exploration of other control architectures
- quantitative characterization of a well known system (yeast)
-


\section{Robustness as an evolutionary design principle}

	\subsection{Ubiquity of noise}	
	
	\subsection{Scale free ratiometric sensing}
	
	\subsection{Selective pressure on negative regulation}
	
		\paragraph{Evidence of further phenotype suppression (papers sent to Rich)}

\section{Harnessing the power of quantitative models}

	\subsection{Demand for accurate measurements}
	
	\subsection{Knowledge at the expense of molecular detail}
		\paragraph{Dynamic model of ratiometric control.}		
		
	\subsection{Generating hypotheses to guide experiments}
		\paragraph{Cell decisions under variable temperature.}		
		\paragraph{Robustness to loss of promoters.}
		\paragraph{Exploration of simpler systems (yeast mating response).}
		
\section{Outlook for studies of cell decisions}

	\subsection{Open questions for further exploration}
		\paragraph{Robustness of cell decisions to other environmental conditions.}	
					
	\subsection{Consolidating knowledge through collaboration}
	
		\paragraph{Grounding theory with experimental validation.}		
		\paragraph{Open source ecosystem.}


















---------- GET CITES FROM NRSA

Robustness is a fundamental organizational principle underlying the evolution of biological systems7. This need for reliability amidst genetic and environmental perturbations is typically assumed to explain the prevalence of redundancy and feedback within gene regulatory networks8. The proposed research poses a complementary explanation for the ubiquity of these regulatory features; redundant negative feedback confers a selective advantage by enabling rapid growth and development. By exploring this hypothesis, the study promises to identify a novel driving force for increased complexity in gene regulatory network topologies. 



Loss of function mutations affecting regulatory mechanisms have been implicated in the emergence of cancer, autoimmune disease, neurological disorders, diabetes, cardiovascular disease, and morphological abnormalities. Understanding when and how regulation fails is vital to controlling its proper function in humans.

1.	Misregulation of gene expression results in human cancer and disease. Genome-wide association studies have shown non-coding regulatory DNA variation to be strongly associated with human cancer and disease1. Understanding how regulatory variants give rise to disease states is vital to the development and implementation of novel medical strategies. These functional relationships have been explored for a number of transcriptional, post-transcriptional, and epigenetic regulators, but in most cases the precise molecular details remain elusive2–5. This study proposes both mechanistic and phenomenological frameworks for studying the emergence of developmental errors when regulation fails to attenuate transient signals. By studying how developmental processes fail when regulation is perturbed in Drosophila and yeast, the proposed research will advance our understanding of how regulation ensures accurate and timely cellular decisions in all animals.  

In the future, model-informed design of engineered repressors could serve as a platform for tuning signaling dynamics in a manner conducive to proper human development and health.

The prevalence of redundant regulation in gene regulatory networks is often ascribed to a need for robustness against genetic and environmental variability8. These experiments motivate an alternate hypothesis; redundant negative feedback facilitates faster growth and development. While intriguing, the data are limited to fractions of animal populations exhibiting abnormal phenotypes and do not address the underlying mechanism. Any approach toward understanding this phenomenon must bridge the gap between gene expression and phenotypic outcomes with sufficient generality to explain the breadth of surveyed systems. The proposed research plan provides an extension to this hypothesis; redundant negative feedback facilitates faster growth and development by coupling developmental gene expression programs to physiological cell state. My hypothesis is predicated on the notion that cells make mistakes when global rates of transcription and translation outpace regulatory networks’ abilities to attenuate transient signals. From this perspective, redundant repressors provide overflow capacity to buffer against increased protein synthesis, ensuring the timely degradation of transiently expressed proteins when growth and development are fast. Together, increased biosynthesis and sufficient negative feedback would enable development to proceed more quickly without incurring additional errors.

The breadth of experiments in Drosophila points toward a dynamic phenomenon agnostic to the molecular detail of repressors and their targets. To demonstrate the generality of my hypothesis, I propose a coarse-grained framework relating expression of a generic protein to population-wide phenotype distributions that does not depend upon the specifics of each developmental system. My goal is to predict how removal of a negative feedback mechanism causes protein levels within affected cells to differ from their unperturbed counterparts. As nuanced changes in expression yield abnormal morphologies in vivo, differences in protein levels manifest as differences in phenotype penetrance25. I aim to quantify both how different a population of cells behaves when negative feedback is removed and how the magnitude of this difference depends upon biosynthesis rates.

The framework suggests the experiments reflect a general principle of dynamic systems; they are more sensitive to perturbation when internal dynamics are fast. In this case, transcription factor activity is more sensitive to changes in regulation when mRNA and protein biosynthesis rates are high.




Together, these results indicate that tightly coordinated competition between Yan and Pnt lies at the heart of neuronal fate commitment. These observations are consistent with the notion that Yan and Pnt compete for occupancy of shared binding sites in the promoter region of downstream effectors of neuronal differentiation(Gabay et al., 1996; O’Neill et al., 1994). Consequently, altering the expression dynamics of either protein is expected to increase the frequency of erroneous fate commitment, and subsequently increase the likelihood of a roughened eye phenotype. 