\chapter{Conclusions}
\label{ch:conclusion}

The preceding chapters explore how the structure and function of gene regulatory networks control cell fate decisions and yield emergent properties at the organismal scale. Each chapter individually advances the fields of systems and developmental biology in unique ways.

Chapter \ref{ch:clones} introduced a computational framework for automated analysis of genetic mosaics; a class of experiments commonly used to probe cell fate decisions \textit{in situ} \cite{Germani2018,Atkins2019}. The framework combines computer vision and statistics to automate the labor-intensive steps of a quantitative work-flow, enabling automated and systematic comparison of cells subject to control and perturbation conditions in an otherwise equivalent background.

Quantitative mosaic analysis itself is not new, nor is it uncommon in high profile publications \cite{Dai2017,Gavish2016,Li2018}. Yet, these studies deploy an irreproducible mix of ad hoc implementations and costly commercial software. Worse still, qualitative analysis pervades less prominent corners of the literature. Contributing a fully automated framework to the open-source ecosystem will make quantitative mosaic analysis accessible to the research community as a whole. A unified framework will also dramatically simplify the reproduction of existing analyses. Chapter \ref{ch:clones} therefore advances the quantification of developmental biology by adding potential for rigor and reproducibility where they are currently lacking.

Chapter \ref{ch:ratio} explored a novel cell fate decision mechanism underlying photoreceptor specification in the \textit{Drosophila} larval eye. Computer vision techniques were used to extract quantitative measurements of Pnt and Yan dynamics from a wealth of confocal microscopy data. Statistical analysis revealed that differentiation is driven by dynamic changes in the ratio between Pnt and Yan, and is agnostic to changes in their absolute concentrations as long as the ratio remains constant. The data therefore provide the first direct evidence that cell fate decisions can be triggered by changes in the relative abundance of separate transcription factors. This finding rebukes the canonical model of photoreceptor specification \cite{Graham2010}. More importantly, it adds a new dimension to our understanding of how multiple transcription factors cooperatively control cell fate decisions, with broad implications for many developmental contexts within and beyond \textit{Drosophila}. 

The ratiometric sensing mechanism identified in Chapter \ref{ch:ratio} adds to a growing body of evidence that cells are able to sense relative changes in the abundance of GRN components \cite{Goentoro2009a,Frick2017}. These discoveries are exciting because relative sensing could potentially isolate cell fate decisions from environmental sources of variation. This is because environmental fluctuations would likely manifest as correlated extrinsic noise that affects all GRN components in a similar manner, causing absolute but not relative concentrations to vary between cells. Relative sensing might then increase fitness in variable environments. 

Regulatory interactions may provide additional layers of stability. Dual-reporter experiments have shown that regulation buffers cell-by-cell differences in yeast gene expression to enhance the precision of decisions to commit to a mating response phenotype \cite{Colman-Lerner2005}. Indeed, Chapter \ref{ch:metabolism} also showed that the microRNA miR-7 buffers Yan expression levels, and by extension R cell fate decisions, against varying biosynthesis capacity. Perhaps future experiments could address how Pnt levels are affected before and after IPC ablation in $yan^{\Delta miR\hyphy 7}$ mutants. 

These observations reflect a central theme of this dissertation; the structure and function of developmental GRNs dictate the robustness of cell fate decisions to environmental variation. Chapter \ref{ch:metabolism} directly embraced this sentiment. It proposed a new theory to explain why the regulatory networks that coordinate cell fate decisions often contain many seemingly redundant repressors acting upon the same target genes. The theory posits that auxiliary negative regulators mitigate erroneous cell fate decisions when cells are rapidly metabolizing, and implies that auxiliary repressors may help GRNs adapt their behavior to environmentally driven variation in cell metabolism. The theory is supported by a diverse collection of experiments in which repressor loss-of-function phenotypes were reversed when biosynthesis rates were slowed. A quantitative modeling framework was used to explore the mechanistic origin of this effect, and theoretically demonstrated that auxiliary repressors could avert erroneous decisions by expanding cells capacity to buffer excess protein expression. Quantitative measurements of transcription factor activity confirmed the hypothesized mechanism in vivo.

Chapter \ref{ch:introduction} introduced the notion that developmental GRNs guide organisms through a tortuous journey from embryo to adulthood. The journey is not a sprint. Rather, cells must carefully navigate the many twists and turns of developmental programs; rapidly synthesizing GRN components when and where they are needed, then degrading them with comparable urgency. Protein synthesis and degradation machineries supply the engine and brakes needed to negotiate these obstacles. Individuals stand to benefit from completing the journey quickly because they are generally vulnerable until adulthood, with little means to defend themselves against predation and other dangers. They could naively swap out the engine for something more potent, but added power escalates the risk of perilous consequences. Instead, they could realize the best of both worlds by simultaneously upgrading the brakes. Similarly, organisms may accelerate development by expanding biosynthesis capacity, but the resultant boost in protein expression can cause erroneous cell fate decisions that lead to the emergence of deleterious phenotypes. By simultaneously incorporating additional repressors, they can tolerate faster metabolic rates without compromising the accuracy of cell fate decisions. The data presented in Chapter \ref{ch:metabolism} demonstrate that auxiliary repressors enable faster development by illustrating the inverse perspective. Repressors were shown to be dispensable when metabolism was slow, much in the way that stock brakes would suffice at low speeds. 

This line of reasoning implies that a novel evolutionary driving force may shape the structure and function of developmental gene regulatory networks. Shorter generational times confer a selective advantage beyond reducing individuals exposure during infancy. They facilitate rapid exploration of the phenotypic landscape, enabling fast adaptation to variable environments. GRNs should therefore be expected to incorporate any topological features that allow development to proceed more quickly. The abundance of seemingly redundant regulation found in developmental GRNs certainly appears to support this hypothesis, and thus reinforce our contemporary understanding that robustness is a fundamental organizational principle underlying the evolution of biological systems \cite{Kitano2004,Stelling2004}. 

The findings also contribute to an emerging view that cells capacity to rapidly adjust protein homeostasis directly affects organismal fitness and health \cite{Visscher2016,Tollerson2018,Burnaevskiy2018}. The assertion is backed by convincing experimental evidence. Burnaevskiy et al used a dual-reporter scheme in \textit{C. elegans} to show that cellular differences in the abundance of protein synthesis machinery manifest in the population-wide penetrance of adult phenotypes. The authors went on to speculate that longevity might be similarly be affected \cite{Burnaevskiy2018}. Tollerson et al showed that Elongation factor P alleviates a translational bottleneck caused by ribosomal queuing in \textit{E. coli}, facilitating adaption to environmentally-driven increases in cell metabolism \cite{Tollerson2018}. Chapter \ref{ch:metabolism} contributes unique evidence that the accuracy of cell fate decisions depends upon cells ability to dynamically balance the push and pull of protein synthesis and degradation. 

%
%\section{ Avenues for further exploration }
%
%The findings presented in Chapters \ref{ch:ratio} and \ref{ch:metabolism} prompt consideration of several new directions for future research.
%
%- Ratio control mechanism?
%- Direct perturbation of Pnt/Yan ratio?
%- Exploration of cis-regulatory syntax
%
%- Relation to temperature - would require knowledge of temperature/metabolism scaling
%- Introduction of simpler model
%- Exploration of tradeoff between synthesis and degradation
%- Potential for a complete 'penetrance theory' framework using moment equations
%- Rigorous exploration of other control architectures: What about promoters?
%- Consideration of other systems (yeast mating response) - potential for quantitative dynamics
%

\section{Lessons for quantitative biology}

This dissertation surveyed developmental cell fate decisions through a quantitative lens. It used numbers and equations to derive nuanced understanding of processes that are notoriously difficult to characterize. Doing so required forcible reconciliation with traditional biological insight. Two valuable lessons were learned in the process whose continued discussion will benefit the field of quantitative biology as a whole.

First, models should fit the data; not the other way around. While it may be possible to coax data into a predefined framework, tailoring one appropriate for the task at hand will generally return more meaningful insight. Mathematical flexibility was a key strength of this dissertation. Each chapter leveraged a customized modeling framework to tease deeper meaning out of experimental data than would otherwise have been possible using conventional techniques. Chapter \ref{ch:clones} deployed a Bayesian statistical framework to infer cell genotypes from an image of clonal marker expression. Biological intuition suggests it should have enforced detection of three distinct components strictly delimited by clonal marker level. Instead, the model was designed to tolerate an arbitrary number of components defined by both clonal marker level and spatial context, which were later aggregated into the three anticipated genotypes. This empirical formulation buffered the uncertainties imparted by expression heterogeneity and imaging noise to improve annotation performance. 

Chapter \ref{ch:ratio} used a model to explore how \textit{cis}-regulatory interactions between Yan monomers bias the transcriptional output of genes simultaneously regulated by Pnt. Conventional wisdom suggests an equilibrium competitive binding model based on the Hill-Langmuir equation would be appropriate \cite{Gesztelyi2012}. Instead, an equilibrium statistical mechanical approach was adapted from earlier work by Hope et al \cite{Hope2017}. Contrary to the empirical strategies used in other chapters, this model substantially increased the resolution of analysis. Doing so provided detailed mechanistic insight into the intricacies of polymerizing transcription factors that would have otherwise been inaccessible (see Fig. \ref{fig:ratio:figS5}C). 

Chapter \ref{ch:metabolism} sought to model the dynamic behavior of GRNs controlling a broad spectrum of cell fate decisions. Biological intuition suggests the first step should have been to identify known regulatory interactions in each system. Such an endeavor was certainly possible for Yan-mediated control of retinal patterning as the pertinent interactions have been studied for decades \cite{Ready1976,Rebay1995,Rohrbaugh2002,Carthew2009}. Chapter \ref{ch:metabolism} instead drew inspiration from control theory to develop a mathematical model strictly concerned with the salient features of pulsatile protein expression dynamics. Namely, the magnitude of induction and timescale of decay. The model neglected the molecular detail of each system, instead favoring a coarse-grained depiction that was much more generalizable. This reductionist approach was vital to the model's success in representing a diverse set of developmental contexts. 

The coarse-grained model deployed in Chapter \ref{ch:metabolism} wholly embraces the second major lesson: The resolution of analysis should match the resolution of the data. Biological networks are complex, often far more complex than we can intuitively comprehend. Their emergent behavior arises from the collective interactions of many components, most of which are typically unknown. These uncertainties make it particularly dangerous to represent systems level behavior as the sum of its parts. Furthermore, predictions made by aggregating interactions that were characterized in isolation are all but meaningless.

Despite its pitfalls, this practice remains tragically common in the literature. It is perhaps most strongly embodied by the abundance of cartoons that purport to depict systems level behavior by assembling a compendium of qualitative regulatory interactions. Figure \ref{fig:ratio:fig7}A provides a convenient example. These cartoons are harmless by themselves, but problems arise when unsuspecting viewers ascribe mechanistic meaning to the individual arrows. There is no unified standard to define what these arrows mean. A single arrow might actually represent an entire sub-network of nonlinear processes. Intermediate chemical species within an arrow might even interact with others in separate arrow drawn elsewhere in the diagram. The rampant ambiguity in these cartoons starkly contrasts the standardized design languages used in engineering \cite{Lazebnik2004}. It also hinders the communication of otherwise outstanding research.

Fortunately, a viable solution to this challenge already exists. Control theory emphasizes an empirical representation of systems-level dynamics that is easily matched to the resolution of available data. It offers a particularly appropriate perspective for rationalizing the behavior of developmental GRNs, as most are inherently localized in time and contain numerous unknown components and interactions. This rationale inspired the conceptual model of ratiometric control used to interpret the Pnt-Yan network analyzed in Chapter \ref{ch:ratio}. Similar reasoning inspired the design of the control theoretic modeling framework used throughout Chapter \ref{ch:metabolism}. The breadth of experiments pointed toward a dynamic phenomenon agnostic to the molecular detail of repressors and their targets. More importantly, among all systems surveyed, only Yan and Sens expression were measurable. The resolution of analysis was therefore matched to the resolution of the data; resulting in the pulsatile expression of a generic protein.

%
%\section{Tools for quantitative biologists}
%
%The research presented in this dissertation spawned several computational tools that may prove valuable to the broader community of quantitative biologists. All of these resources have been made freely available online under open license for unrestricted use and future development. They are published alongside \textbf{FlyEye Silhouette}: \url{http://silhouette.amaral.northwestern.edu}, a GUI-based MacOS application for segmentation, quantification, and annotation of cell nuclei in the \textit{Drosophila} eye imaginal disc that was developed by Helio Tejedor in the Amaral lab at Northwestern University. The list below describes each of the new tools and their high level functions. All are accessible via GitHub repositories mirrored between both \href{https://github.com/sebastianbernasek/}{my personal account} and the \href{https://github.com/amarallab}{Amaral} and \href{https://github.com/bagherilab}{Bagheri} lab accounts. These repositories contain high level API documentation in addition to a series of Jupyter notebooks that walk the user through a series of usage examples. 
%
%
%\textbf{NOTE}: See Appendix \ref{appendix:resources} for all experimental data and another set of Jupyter notebooks walk the user through reproducing all of the results and figures presented in this dissertation.
