\chapter{Introduction}

The natural world presents a stunning variety of multicellular organisms. We tend to distinguish them by the physiological traits that define their stereotyped identities. For instance, we recognize penguins by their unusual stature, black and white fur, long beak, and impressive ability to wobble around on ice. These features are a culmination of many complex cellular processes collectively known as \emph{development}.

Development begins with a single cell, with subsequent growth and division events driving progression toward adulthood. Cells acquire increasingly specific roles and functions as growth proceeds, ultimately giving rise to stereotyped adult morphologies. This process, known as lineage restriction, demands that each cell decides to adopt the correct fate at the appropriate time and place.

Cell fate decisions are remarkably robust, yielding consistent phenotypes amidst the vast array of conditions encountered in natural environments. They are so reliable that we often take them for granted. For instance, it is difficult to imagine a scenario in which we might mistake another human for a penguin. However, they can and do exhibit minor variation that can have dire consequences for human health. Researchers therefore continue to study these complex processes in the hope that they might one day be able to control their behavior; either to exploit them in novel biotechnologies, or rectify the diseases that arise when they fail. Research efforts are predominantly motivated by two fundamental questions. First, how do cells make decisions? Second, how do they make them reliably? 

This dissertation addresses subtle aspects of both questions by combining chemical engineering, computer vision, and statistics to extract meaningful insight from experimental data. The remaining sections of this chapter serve to prime the reader with the context needed to situate the presented findings within the broader literature. 

\section{Molecular origins of cell fate decisions}

Early experiments in a variety of model organisms demonstrated that cell fate decisions are encoded in the genome. Genetics were therefore believed to provide a predefined road map for the tortuous journey from embryo to adulthood, inspiring researchers throughout the twentieth century to probe the roles of individual genes in coordinating adult phenotypes. Most of their efforts embraced a common philosophy; break something and see what happens. Nowhere is the legacy of this approach more pronounced than in the common fruit fly, \textit{Drosophila melanogaster}, whose genes are predominantly named after the phenotypes that emerge in their absence. Removing \textit{eyeless}, \textit{wingless}, or \textit{hairy} may now seem rudimentary, but these genetic perturbations, and others like them, were vital to the discovery of genetic components and architectures that dictate cell fate decisions in all organisms. They revealed that some genes confer pleiotropic functions across several stages of development, while others are limited to a single context. They also showed that some genes are only essential for proper development when others are absent, rendering them redundant under normal conditions. Moreover, \emph{Drosophila} continues to be a prominent model system for studying developmental processes today, owing to its conveniently short life cycle, wealth of prior knowledge, and deep library of available genetic machinery \cite{Beira2016,Enomoto2018,Germani2018}.

Among these tools, gene-specific reporters have augmented traditional genetic perturbations by providing localized readouts of transcript and protein abundance. Researchers can now break something and see what happens to specific components of the developmental program. Reporters led to the discovery of transcription factors; proteins that bind the promoter region of other genes in order to modulate their expression. Multiple transcription factors may interact with each other, allowing for the assembly of gene regulatory networks (GRN) that integrate upstream signaling cues to elicit specific changes in gene expression. 

GRNs thus arm cells with a chemical mechanism to orchestrate cell fate decisions in space and time. A prominent example occurs during the during the earliest stages of \textit{Drosophila} embryogenesis, where spatial morphogen gradients drive variegated expression of four Gap genes. The expressed proteins trigger subsequent developmental events in a concentration-dependent manner, inducing localized cascades of GRN activity that ultimately give rise to distinct organs. The early landscape of Gap gene expression thereby defines a template for later stages of patterning.

The spatial resolution of GRN activity is thought to be further refined over time. This assertion is in part based on experimental studies of retinal patterning in the \textit{Drosophila} eye, a setting with an enduring experimental legacy that continues to garner attention \cite{Beira2016}. This is in part because the temporal history of cell fate decisions is encoded in static images of the eye imaginal disc. During the third larval instar, a wave of differentiation steadily progresses across a disordered pool of multipotent cells \cite{Ready1976a,Tomlinson1987a}. Differentiating cells propagate this morphogenetic furrow (MF) by relaying extracellular cues downstream. Decades of experiments have steadily revealed how cells interpret the signals and commit to particular fates \cite{Voas2004}. Fate-specific reporters have shown that R8 photoreceptor neurons are recruited from pools of multipotent cells at the leading edge of the furrow \cite{Jarman1994}. The first cell to differentiate uses paracrine signaling to inhibit differentiation among its neighbors, resulting in the repeated pattern of regularly-spaced R8 cells that gives rise to the crystalline lattice structure of the adult eye \cite{Frankfort2002a}. The process dynamically transforms a disordered field of cells into an ordered template for subsequent stages of patterning, thus refining the spatial resolution of the developing eye field.

Experiments suggest the race to an R8 fate is non-deterministic. The R8 cell fate transition is driven by the expression of Atonal, a transcription factor initially induced in all cells along the leading edge of the furrow \cite{Jarman1994,Baker1997,Hsiung2002}. Stochastic differences in cells reception of signaling cues yields variation in Atonal levels and, consequently, cells propensity to adopt an R8 fate \cite{Baker1990,Gavish2016}. The eventual R8 cell therefore does not appear to be predetermined. Equivalent mechanisms control sensory bristle formation and the choice between epidermal and neural fates in the neuro-ectoderm \cite{Ghysen1993,Simpson1997}.

These three examples showcase another core feature of GRNs: their components are heterogeneous, even within the same developmental context. Variation may be attributed to both intrinsic and extrinsic sources of noise, each of which can manifest in the population-wide penetrance of phenotypes \cite{Raj2010,Paulsen2011,Burga2011,Colman-Lerner2005}. The retinal patterning example suggests some systems even leverage heterogeneity, using GRNs to amplify and reinforce stochastic fluctuations to limit signal responses to a random subset of cells \cite{Baker1990,Ghysen1993,Simpson1997}. More recently, Pel'{a}ez et al proposed a similar mechanism to explain why subsequent R cell fate transitions coincide with rapid increases in transcription factor expression heterogeneity \cite{Pelaez2015a}.

These experiments and others like them form our contemporary systems-level view of development, in which complex networks of regulatory interactions guide cells toward the appropriate fates by refining their stochastic behavior over time. However, it is unknown precisely how cell fates are resolved from the blueprints encoded in GRN activity. It is equally unclear how cells integrate stochastic inputs to execute reliable cell fate decisions. These ambiguities persist despite the wealth of experimental data generated throughout the past century. New approaches are therefore needed to unravel the complexities of GRNs that govern cell fate decisions.

\section{The need for quantitative analysis}

Experimental perturbations continue to supply some of the most potent tools in the arsenal available to researchers, but it has become clear that unified and systematic analysis frameworks are needed to tease apart the complex interactions that govern cell fate decisions \cite{Lazebnik2004,Oates2009}. Moreover, attempts to either rectify faulty decision mechanisms or engineer new ones, such as for cancer treatment or the generation of induced pluripotent stem cells, demand predictive models backed by quantitative data \cite{Hornberg2006}.

The focus has therefore gradually shifted from break something and \textit{see} what happens to break something and \textit{measure} what happens. The transition is supported by simultaneous advances in the resolution with which we can quantify GRN activity during development \cite{Sbalzarini2016}. The state of the art has progressed from aggregate measurements of transcript and protein content across entire tissues, to in vivo measurements of individual cells, to in situ quantification of single-cell expression dynamics. These advances bestow mathematical models with the quantitative insight required to generate testable predictions for further experimental validation.

Quantitative measurements and analysis have already begun to illuminate cell fate decisions and challenge canonical theory \cite{Frick2017,Petkova2019,Wolff2018}. Petkova et al recently used information theory to show that the Gap proteins provide sufficient spatial context to uniquely define the eventual fates of all cells \cite{Petkova2019}. They further argued that downstream GRNs likely decode these signals with near-optimal efficiency, implying that several cell fates may actually be determined during the earliest stages of embryogenesis. The study refutes the established notion that cell fates are spatially resolved as development proceeds, and illustrates the novel insight that may be derived from quantitative data.

Sustained prevalence of subjective analysis in the literature indicates that researchers have not universally adopted quantification. One plausible explanation is that quantification often demands computational proficiency that falls beyond the scope of many experimental labs. Interdisciplinary collaborations are becoming more frequent and should help alleviate this challenge, but they do not provide a permanent solution. These studies could benefit from the introduction of open-source automated analysis software platforms analogous to those that have revolutionized other subdisciplines of biology \cite{Aghaeepour2013,Chen2015,Pyne2009,Bernstein2008,Hellemans2007,Langmead2012,Trapnell2009,Costes2004,Kelley2015,Carpenter2006,Paintdakhi2016,Schindelin2012,Sommer2011}. For example, without the support of automated sequence alignment software, RNA-seq would be inaccessible to all but a few researchers with extensive programming and statistical modeling experience. Similar tools are therefore needed to support quantitative measurements of gene expression during development \cite{Jug2014,Sbalzarini2016,Schindelin2015}, thus lowering the barrier to adoption of data-driven modeling of cell fate decisions.

\section{Mathematical modeling of cell fate decisions}

Mathematical models have reinvigorated the study of cell fate decisions in a broad variety of contexts. Most modeling efforts fall into one of two categories; those that use a data-driven approach to recapitulate molecular mechanism, and those that provide a sparse representation of systems-level behavior. 

The first class of models strive to parameterize specific biomolecular interactions by fitting a model directly to data. They typically describe the time-evolution of transcripts and proteins using systems of coupled ordinary differential equations (ODEs) reminiscent of those familiar to chemical engineers and ecologists. Despite the illusion of mechanistic detail, these models still deploy a healthy dose of abstraction. None of the commonly used rate represent true elementary reactions, instead opting for empirical representations such as linear degradation and cooperative binding kinetics. Nevertheless, many novel GRN behaviors and functions have been elegantly proposed and tested in this manner \cite{Barkai1997b,Yu2008a,Paulsen2011}. 

The second class of models forego molecular detail in favor of a coarse-grained representation of a particular phenomenon. These approaches provide a powerful means to identify, characterize, and predict behaviors that span a broad variety of model systems and developmental contexts. Among the common modeling frameworks, control theory has proven particularly fertile for generating and testing hypotheses related to GRN dynamics. Bacterial chemotaxis offers a compelling example in which experimentally derived molecular models were supplanted by a simple integral control framework \cite{Barkai1997b,Alon1999,Yi2000,Muzzey2009}. Analogous strategies have drawn inspiration from several disciplines to identify numerous functions of GRNs \cite{Ma2009,Colman-Lerner2005,Rahimi2016,Benzinger2018,Adler2018,Yordanov2018}.

Coarse-grained models have contributed a particularly unique perspective as to how cell fates are resolved from spatiotemporal signaling cues. These problems would otherwise be intractable due to the many complex transport processes that shuttle signaling molecules between neighboring cells. Lubensky et al used a comparatively simple reaction-diffusion approach to model pattern formation in the larval eye. They showed that inductive signaling cues could drive cell-autonomous positive feedback to account for the emergence of a hexagonal lattice of R8 cells, as well as the previously-unexplained striped pattern observed in some mutants \cite{Lubensky2011}. Gavish et al used an even simpler model to show that an additional inhibitory signal is necessary to stabilize retinal patterning against minor fluctuations in cells spatial arrangement. The authors then used an experimental technique known as quantitative mosaic analysis to identify the unknown diffusible inhibitor \cite{Gavish2016}.

Mathematical models have also shone light on the regulatory interactions that implement cell fate decisions within individual cells. One study explored how cells generate all-or-none responses to morphogen gradients in the \textit{Drosophila} ventral ectoderm \cite{Melen2005}. An ultrasensitive response mechanism was proposed to dictate the expression of Yan, a transcriptional repressor known to impede cell fate transitions \cite{Lai1992a,Rogge1995,Rebay1995}. A later study proposed that Yan plays a different role in the larval eye, instead forming a bi-stable switch through reciprocal antagonism with a transcriptional activator named Pointed (Pnt) \cite{Graham2010}. The authors used a detailed model to demonstrate that the decision to adopt an R cell fate could be triggered by an irreversible transition between two stable states; one characterized by high Yan and low Pnt expression, and the other by low Yan and high Pnt expression. The authors proposed that the switch is flipped via sustained induction of Pnt expression. Both transcription factors are known to be subject to several seemingly redundant sources of negative feedback. The model rationalized the purpose of these inhibitors by postulating that they enforce separation of the two stable states. Shwartz et al extended the model to include autoregulatory interactions that help flip the switch by converting transient inductive signals into sustained Pnt expression \cite{Shwartz2013}. 

Curiously, a qualitative report later showed that Yan and Pnt are co-expressed in several developmental contexts \cite{BoisclairLachance2014}. Pel\'{a}ez et al then published quantitative measurements that show Yan exhibits apparently mono-stable expression dynamics in the larval eye \cite{Pelaez2015a}. These two studies fundamentally contradict the existing model, prompting renewed interest in both Yan and Pnt. Namely, how do they cooperate to mediate cell fate decisions? How do they do so reliably? And why are they subject to so much negative feedback?

\section{Roles for negative feedback in GRNs}
 
Theory supports many potential uses for negative feedback in developmental GRNs \cite{Freeman2000}. Perhaps the most obvious application is the maintenance of cell states --- that is, rejecting exogenous disturbances and driving cells toward a desired set point \cite{Alon2007,Behar2007,Yi2000}. Negative feedback has also been shown to improve information transmission by linearizing input-output relationships and expanding dynamic range in several developmental signaling cascades \cite{Bhalla2002,Cheong2011,Paulsen2011,Yi2003,Yu2008a}. Yi et al. used a FRET-based reporter of the G\textalpha-subunit to demonstrate dose-response alignment of G-protein signaling activity at the highest level of the yeast mating response pathway \cite{Yi2003}. Yu et al. later quantified pFus3 activity to show that negative feedback further increases signal fidelity throughout much of the downstream pathway \cite{Yu2008a}. These functions mimic important uses of negative feedback in human-engineered systems \cite{Khammash2016}.

Negative feedback is also vital to the reliable execution of cell fate decisions. Rahimi et al showed that WntD-mediated negative feedback enhances the precision of morphogen gradients in the ventral domain of the \textit{Drosophila} embryo \cite{Rahimi2016}. Paulsen et al. showed that the fidelity of BMP4 signaling in the \textit{Xenopus} embryo is improved by concomitant expression of a repressor that suppresses transduction of extrinsic noise. Eliminating the repressor leads to increased variation of pathway outputs, as well as the cell fate decisions they control \cite{Paulsen2011}. These studies provide direct experimental evidence that negative feedback can suppress phenotypic variation. 

Experiments in \textit{Drosophila} indicate that short non-coding transcripts, called microRNAs, confer a similar function by buffering developmental processes against both environmental and genetic variability \cite{Cassidy2016a,Cassidy2013,Li2009b,Ebert2012}. Li et al. studied the effect of perturbing miR-7 activity during sensory organ development, and found that the microRNA stabilizes both gene expression and fate commitment decision against fluctuating environmental conditions \cite{Li2009b}. Cassidy et al. used directional selection to quantify the heritability of bristle formation defects in miR-9a mutants, revealing that microRNAs can also suppress the influence of intrinsic genetic variation \cite{Cassidy2013}. The same group later showed that miR-9 suppresses the penetrance of genetic variants deleterious to viability at elevated temperatures \cite{Cassidy2016a}. Negative feedback mediated by microRNAs has therefore been shown to directly affect the evolutionary fitness of adult organisms by promoting robustness against varying environmental conditions. Precisely how this effect is achieved remains unknown.

\section{Evolutionary drivers of robust cell fate decisions}

Studies of developmental robustness are often traced back to C.H. Waddington’s claim that “there is scarcely a mutant that is comparable in constancy with the wild type” \cite{Waddington1942}. His comment reflects the consistency of adult phenotypes amidst modest levels of genetic and environmental variation, which gives way to variation when severe perturbations are introduced \cite{Bateman1959a,Rendel1959,Rendel1966a,Scharloo1991}. Robustness has since been attributed to individual genes and their products \cite{Dun1958,Gibson1996,Rutherford1998}, as well as the systems-level architectures that they comprise \cite{Rutherford1998,Paulsen2011,Li2009b,Eldar2002,Denby2012,Cassidy2013,Cassidy2016a}. A handful of conserved regulatory motifs, and the strategies they implement, are now understood to pervade most developmental processes \cite{Freeman2000,Hartman2001,Alon2007,Marciano2014}. Robustness has thus come to be accepted as a fundamental organizational principle underlying the evolution of all biological systems \cite{Kitano2004,Stelling2004}. That is, selection for genotypes that confer robustness is widely assumed to shape the evolution of gene regulatory network topologies. 

The specific evolutionary drivers for increased robustness of GRNs remain unclear \cite{Siegal2014}. Citing phenomenological examples, Waddington attributed robustness to evolutionary selection for optimal traits. More recently, Siegal et al. argued that developmental robustness can arise without stabilizing selection for any particular phenotype \cite{Siegal2002}. They used a gene interaction network model to demonstrate that robustness is an emergent property of complex networks subject only to selection on the basis of their own functional stability. The same authors used a an \textit{in silico} evolution approach to show that most genes in developmental networks buffer phenotypic variation, indicating that robustness may be an emergent byproduct of GRN architecture \cite{Bergman2003}. Computational studies have also questioned the notion that global GRN properties are a consequence of adaptive evolution. Rather, local properties of genetic circuits could drive the emergence of macroscopic features, such as those that ensure robust development \cite{Lynch2007,Wagner2003}.

When a gene or its products are absent or fail to perform their functions, the simplest contingency is to have others ready to compensate. It consequently seems intuitive that cells could incorporate functional redundancy to guarantee reliable cell fate decisions \cite{Hartman2001,McAdams1999}. One computational study showed that genetic redundancy is particularly evolutionarily stable in developmental systems. By simulating functionally overlapping pairs of genes, the authors demonstrated that increasing the probability with which each gene independently fails to carry out its function places increasing selection pressure upon the pair as a whole \cite{Nowak1997}. They reasoned that the resultant increase in evolutionary stability explains the prevalence of redundancy in developmental systems, where failure rates are high and erroneous cell fate decisions yield deleterious phenotypes. Their suggestion is consistent with extensive documentation of functional redundancy in developmental systems \cite{Kitano2004}. 

Redundancy could arise through gene duplication or convergent evolution. A. Wagner argued against the former by leveraging gene sequence and expression data in yeast to show that the phenotypic severity of loss-of-function mutations does not correlate with the availability of alternate functionally equivalent genes. He concluded that functional redundancy is not the primary source of robustness against genetic variation, instead favoring epistatic interactions between otherwise unrelated gene products \cite{Wagner2000}. Combined, these studies suggest the prevalence of functional redundancy in GRNs may be attributed to as-of-yet unknown forces.
