\includegraphics[width=6.50000in,height=4.95347in]{media/image1.png}

\textbf{Dynamics of Pnt and Yan expression during eye development.}

(A) Diagram of the \emph{pnt} locus encoding the Pnt-P1 and Pnt-P2 protein isoforms. The isoforms share an ETS DNA-binding domain (red) but are distinguished by the presence of a SAM domain (blue) within Pnt-P2. Green arrows at the C-termini depict insertion sites of GFP in the \emph{pnt-gfp} allele, black arrows at the N-termini depict insertion sites of the pnt\textsuperscript{HS20} and pnt\textsuperscript{1277} enhancer-trap alleles. Adapted from Shwartz et al. (2013).

(B) Differentiation is initiated in the developing eye by the MF, which moves across the eye epithelium from posterior to anterior (white arrow). On the furrow's posterior side, G1-arrested progenitor cells differentiate (light blue). Formation of regularly spaced R8 photoreceptors (red dots) precedes recruitment of additional R cell types (yellow dots). On the anterior side, progenitor cells are still proliferating (dark blue). Axis refers to time elapsed since fertilization. Adapted from Peláez et al. (2015).

(C) Top, cartoon of an apical view of the sequential differentiation of eight R cell types from multipotent progenitor cells (grey) and their relative positions within a single ommatidium. Arrows denote signals transmitted from the R8 to nearby cells. Bottom, a cross-section view through an eye disc, showing the epithelial constriction that marks the MF (boxed region) and then the relative nuclear positions of progenitors (grey) and specified R cells (various colors). Together, the stereotyped features depicted in this schematic enable unambiguous identification of each cell type as ommatidial assembly proceeds. Adapted from Peláez et al. (2015).

(D) Maximum intensity projection of Pnt-GFP fluorescence in an imaginal disc fixed \textasciitilde{}100 h after fertilization. Right panel corresponds to the region enclosed by dashed white lines in the left panel. Morphogenetic furrow (orange arrow) precedes the first and second stripes of strong Pnt-GFP expression (black lines, labeled 1 and 2).

(E) Pnt-GFP expression in progenitor cells. Grey points are individual cells, solid line is the smoothed moving average. Orange shading indicates the first and second stripes of Pnt-GFP expression.

(F, G) R cell recruitment from the (F) first and (G) second pulses of Pnt-GFP expression. Solid lines and shaded regions denote moving averages and their 95\% confidence intervals.

(H, I) Confocal images of (H) Pnt-P1 and (I) Pnt-P2 enhancer-trap co-expression with Pnt-GFP. Orientation is consistent with panel D. Morphogenetic furrow (orange arrow) precedes first and second stripes of Pnt-GFP induction (black lines, labeled 1 and 2). In merged images, Pnt-GFP is green and enhancer-trap expression is magenta.

(J, K) Measured (J) Yan expression and (K) log\textsubscript{2}-transformed Pnt-GFP to Yan ratios in progenitor cells. Grey points are individual cells, solid line is a smoothed moving average.

\textbf{\\
}

\includegraphics[width=6.50000in,height=5.98056in]{media/image2.png}

\textbf{Pnt-to-Yan ratios differ between cellular states.}

(A-F) Measured (A, B) Pnt-GFP expression, (C, D) Yan expression, and (E, F) Pnt-GFP to Yan ratio dynamics in R2/R5 (blue) and R1/R6 (red) cells. Progenitors are grey. Solid lines are smoothed moving averages across 250 and 50 samples for progenitor and R cells, respectively. Yellow shading indicates time spanned by young R cells.

(G) Comparison of Pnt-to-Yan ratio levels between young R cells (color filled boxes) and their concurrent progenitors (grey filled boxes). Colors denote R cell types. For each R cell type, the ten earliest identifiable R cells in each disc were designated as young R cells. Progenitor cells that fall within the time window spanned by these young R cells were designated as concurrent progenitors. Asterisks indicate significant differences (KS 2-sample test, p\textless{}0.001).

(H, I) Joint distributions of Pnt-GFP and Yan protein levels for young (H) R2/R5 and (I) R1/R6 cells. Progenitor cells concurrent with the corresponding young R cells are shown in grey. Black line denotes the median Pnt-to-Yan ratio among the progenitor cells shown.

\textbf{\\
}

\includegraphics[width=6.50000in,height=2.37986in]{media/image3.png}

\textbf{Cooperative DNA-binding sensitizes transcriptional output to the Pnt-to-Yan ratio. }

(A) Cartoon of competition between Pnt and Yan for occupancy of mutual binding sites in the absence of Yan polymerization.

(B) Overall binding site occupancy as a function of transcription factor abundance in the absence of Yan polymerization. Color scale reflects overall Pnt site occupancy. A diverging scale was used because all sites are fully saturated at total transcription factor concentrations above 1 nM. Under the range of conditions shown, this implies that Yan occupies all sites left vacant by Pnt. Simultaneous proportional increases in absolute abundance of both species have minimal impact on Pnt occupancy (dashed arrow), while varying ratio confers gradual change (solid arrow).

(C) Pnt occupancy of individual binding sites as a function of Pnt-to-Yan ratio in the absence of Yan polymerization. Contours correspond to a vertical path traversed across panel B at a fixed Yan concentration of 50 nM. All binding sites behave similarly.

(D) Cartoon of competition between Pnt and Yan for occupancy of mutual binding sites when Yan polymerizes via its SAM domain.

(E) Overall binding site occupancy as a function of transcription factor abundance when Yan polymerizes via its SAM domain. Color scale and arrows retain their meaning from panel B.

(F) Pnt occupancy of individual binding sites as a function of Pnt-to-Yan ratio when Yan polymerizes via its SAM domain. Contours correspond to a vertical path traversed across panel E at a fixed Yan concentration of 50 nM. Line colors reflect binding site positions within the \emph{cis}-regulatory element. Sites at intermediate distances from the strong ETS site (green lines) transition at higher ratios than those nearest and furthest from the strong ETS site (blue and yellow lines).

\textbf{\\
}

\includegraphics[width=6.50000in,height=5.17014in]{media/image4.png}

\textbf{The Pnt-to-Yan ratio is stabilized against varying Pnt and Yan concentrations in progenitor cells.}

(A-D) Moving averages of (A, B) Pnt-GFP and (C, D) Yan levels in progenitor and differentiating cells with one (A, C) versus two (B, D) copies of the \emph{pnt-gfp} gene. Measurements used DAPI to mark nuclei. Colors denote cell type. Shaded regions are bootstrapped 95\% confidence intervals for the moving average.

(E) Comparison of Pnt-to-Yan ratios between progenitor cells with one versus two copies of \emph{pnt-gfp} during cell fate transitions. Colors denote cell fate transition periods for each R cell type. These time periods are defined in each disc by the times spanned by the first ten identifiable R cells. The concurrent progenitor cell populations are selected from these time windows. Light grey filled boxes denote 1x \emph{pnt-gfp}, dark grey filled boxes denote 2x \emph{pnt-gfp}. Pnt-to-Yan ratios in progenitor cells are indistinguishable between gene dosages during R8, R2/R5, and R1/R6 cell fate transitions (KS 2-sample test, p\textless{}0.001).

(F-G) Confocal image slice of progenitor nuclei in a disc containing loss-of-function \emph{yan} clones. RFP fluorescence marks wildtype \emph{yan}.

(H) Quantitative comparison of Pnt-GFP expression between \emph{yan} genotypes. Progenitor cells were assigned \emph{yan} genotypes based on measured RFP level, and Pnt-GFP levels were corrected to account for fluorescence bleed-through (see \emph{Methods}). Pnt-GFP levels decrease when no gene copies of \emph{yan} are present (Mann-Whitney \emph{U} tests). Red dots denote the median of each distribution, thick grey lines denote the interquartile range.

\includegraphics[width=6.22222in,height=5.54167in]{media/image5.png}

\textbf{Notch signaling lowers the Pnt-to-Yan ratio in progenitor} \textbf{cells.}

(A) Visualization of relative Pnt and Yan expression in progenitor cells in region 2 when Notch signaling is active (left panel) and restricted (right panel). Color scale reflects the difference between Pnt-GFP and Yan fluorescence. Black lines denote periods of elevated Pnt-GFP expression. See methods for details on post-processing of images.

(B) Visualization of relative Pnt and Yan expression in progenitor cells during the first wave of cell state transitions. Black arrow marks the morphogenetic furrow. Gold arrows annotate clusters of elevated ratio.

(C-E) Quantification of spatial periodicity in the Pnt-to-Yan ratio among progenitor cells immediately posterior to the MF when Notch signaling is active. (C) Spatial correlation functions for progenitor cells in four eye discs. Black lines show the moving average pairwise correlation of Pnt-to-Yan ratios between cells as a function of their separation distance along the dorso-ventral axis. Oscillatory forms indicate alternating regions of similar and dissimilar behavior relative to the population-wide mean. Lines are obtained via first-order Savitzky-Golay filtration with a window size of 50. Shaded region shows a bootstrapped 95\% confidence interval for the moving average. Cell counts are annotated above each correlation function. Red lines are the expected outcome for random expression (no pattern). (D) Normalized Lomb-Scargle periodograms for each disc. Spectra are constructed from individual progenitor cell measurements for periods ranging 50 to 200 px. Grey lines denote spectral power attributed to each oscillation period. Dashed red significance thresholds are obtained by bootstrap resampling the ratio intensities. Asterisks denote signal frequencies exceeding the confidence threshold. In all discs, a pattern in Pnt-to-Yan ratios repeats on a length scale of 73-82 px when Notch signaling is active. (E) Distribution of dorso-ventral separation distances between adjacent R8 neurons within a single column of ommatidia within each disc. Mean values are comparable to the detected oscillation periods.

(F,G) No periodicity is detected above the significance threshold when Notch signaling is restricted.

\textbf{\\
}

\includegraphics[width=5.16667in,height=3.05556in]{media/image6.png}

\textbf{Ras signaling elevates the Pnt-to-Yan ratio in progenitor} \textbf{cells.}

(A-C) Effects of constitutive Ras signaling on (A) Pnt-GFP, (B) Yan, and (C) Pnt-to-Yan ratio dynamics in progenitor cells. Lines are moving averages across 250 sequential cells. Shaded regions are bootstrapped 95\% confidence intervals for the mean. Solid lines and grey shading denote wildtype controls. Dashed lines and red shading denote constitutive Ras signaling by \emph{Sev\textgreater{}Ras\textsuperscript{V12}}. Black bars denote periods of elevated Pnt-GFP expression. We previously reported a modest increase in the duration of Yan-YFP expression in \emph{Sev\textgreater{}Ras\textsuperscript{V12}} mutant progenitor cells (Peláez et al. 2015), but this difference was not detected using the Yan antibody.

(D, E) Comparison of Pnt-to-Yan ratios between wildtype and \emph{Sev\textgreater{}Ras\textsuperscript{V12}} progenitor cells concurrent with the ectopic differentiation of (D) unidentified R cells and (E) R7 cells in \emph{Sev\textgreater{}Ras\textsuperscript{V12}} discs. Markers denote the first 25 supernumerary R cells.

\textbf{\\
}

\includegraphics[width=3.13889in,height=2.79167in]{media/image7.png}

\textbf{Conceptual models for regulation of the Pnt-to-Yan ratio. }

(A) Cartoon of qualitative regulatory interactions suggests Pnt and Yan protein levels are coupled by reciprocal positive feedback (solid lines), while Notch and Ras signaling adjust the Pnt-to-Yan ratio by modulating the levels of each protein (dashed lines).

(B) Block diagram of ratio control in the Pnt-Yan network. Lines represent values, rectangles indicate functions, and the circle is a comparison point. The Pnt-to-Yan ratio is compared against a basal reference value, and the difference is fed into a regulatory network that acts to drive the ratio back toward the reference value. Extracellular signals transiently perturb the ratio by modulating Pnt or Yan protein levels (dashed black line), or set the ratio by adjusting the reference value (dashed red line).

\textbf{\\
}

\includegraphics[width=6.50000in,height=4.53542in]{media/image8.png}

(A,B) Adult eyes of flies carrying (A) one or (B) two copies of the recombineered \emph{pnt-gfp} transgene under a \emph{pnt} null mutant background. Note the wildtype retina patterning under both rescue conditions.

(C) Maximum intensity projections of Pnt-GFP fluorescence across layers spanning multipotent cells, differentiating R cells, and differentiating cones. White arrow denotes morphogenetic furrow. Black bars denote first and second periods of elevated Pnt-GFP expression.

(D) Simultaneous Pnt-GFP (green) and Yan (magenta) expression dynamics in progenitor cells. Lines are smoothed moving averages across 500 sequential progenitors, shaded regions are bootstrapped 95\% confidence intervals for the mean. Arrows indicate the times at which local maxima occur.

\textbf{\\
}

\includegraphics[width=6.50000in,height=5.16319in]{media/image9.png}

(A,B) Measured expression dynamics for all annotated cell types. Solid lines are moving averages across 250 and 75 sequential cells for progenitors and differentiating cells, respectively. Shading denotes bootstrapped 95\% confidence interval for the moving average. Colors denote cell type.

(C-E) Expression dynamics and joint Pnt-Yan distributions for differentiating R8, R3/R4, and R7 cells. Joint distributions are limited to progenitor and R cells drawn from the shaded yellow region spanning the first ten R cells of the specified type in each disc. Solid lines are smoothed moving averages across 250 and 50 samples for progenitor and R cells, respectively.

(F-G) Expression dynamics and joint Pnt-Yan distributions for differentiating C1/C2 and C3/C4 cone cells.

\textbf{\\
}

\includegraphics[width=2.37500in,height=4.41667in]{media/image10.png}

(A-C) Dynamics of expression variability in progenitor and R8, R2/R5 and R1/R6 cells. Heterogeneities of (A) Pnt expression, (B) Yan expression, and (C) the log\textsubscript{2}-transformed ratio are estimated by de-trending fluctuations about a moving average of 250 sequential cells. Lines are moving averages of 250 sequential fluctuations, shaded regions are bootstrapped 95\% confidence intervals for the moving average. Colors denote cell type.

\textbf{\\
}

\includegraphics[width=3.61111in,height=1.87500in]{media/image11.png}

(A) Schematic of a simple two-species competitive binding model.

(B) Theoretical Pnt site occupancy as a function of transcription factor abundance. Equivalent binding affinities are used for illustrative purposes. Simultaneous proportional increases in absolute abundance of both species have minimal impact on Pnt occupancy, while varying ratio confers maximal change.

\textbf{\\
}

\includegraphics[width=6.41667in,height=5.19444in]{media/image12.png}

(A) Summary of thermodynamic interactions within one microstate of a cis-regulatory element containing one ETS site and two non-ETS sites. Solid black lines represent individual binding sites. Green and magenta rectangles denote Pnt and Yan molecules. Example thermodynamic potentials of strong ETS-binding, weak non-ETS binding, and polymerization interactions are denoted by \emph{α\textsubscript{Pnt}}, \emph{β\textsubscript{Yan}}, and \emph{γ\textsubscript{Yan}}, respectively. For this microstate, \emph{a\textsubscript{P}(k)}=1 and \emph{a\textsubscript{Y}(k)}=2. (B) Enumeration of all possible microstates for a cis-regulatory element of length 3 in which only the first site carries the ETS designation. Solid black lines denote binding sites, green and magenta rectangles denote bound Pnt and Yan molecules. The cumulative thermodynamic potentials of each microstate, \emph{ΔG\textsubscript{k}}, are listed beside each graphical depiction. (C) Relative thermodynamic contributions of binding site affinity versus polymerization to microstate statistical frequencies as a function of Pnt and Yan concentration. For each point in the plane, influence of site affinity was calculated by weighting the sum of all ETS and non-ETS thermodynamic potentials for each microstate by the statistical frequency of the corresponding microstate. The influence of polymerization was analogously determined. The shown color scale reflects the relative magnitude of these two summations, normalized by limits of zero and complete polymerization.

\textbf{\\
}

\includegraphics[width=6.50000in,height=4.74931in]{media/image13.png}

(A, B) Confocal images of (A) progenitor cells and (B) photoreceptors and cone cells in \emph{yan} null and heterozygote clones. Regions of Ubi-mRFPnls expression are manually labeled by \emph{yan} genotype. White text indicates regions of reduced \emph{Yan} abundance, red denotes wildtype. DAPI visualizes all nuclei.

\textbf{\\
}

\includegraphics[width=5.30556in,height=7.38889in]{media/image14.png}

(A, B) Maximum intensity projections across confocal layers spanning progenitor cells when Notch signaling is (A) active and (B) restricted. Middle and right panels show Pnt-GFP and Yan expression, left panel shows merge in which Pnt-GFP is green and Yan is magenta. Black bars denote first and second periods of elevated Pnt-GFP expression. Dashed yellow line indicates crop boundary used to construct Figures 5A and 5B.

(C) Distances between adjacent ommatidial columns in Notch mutant discs. Left panel: Procedure used to estimate inter-column distance. Neighboring R8 cells are identified by Delaunay triangulation, with an added constraint that edges must fall within 30 to 60 degrees of the anterior-posterior axis. The inter-column distance is estimated by averaging the anterior-posterior distance between neighbors (solid orange line). Right panel: Inter-column distances are more variable when Notch signaling is restricted (red) than under wildtype (grey) conditions. Black bars denote median, numbers above violins indicate the number of neighboring R8 cells. High variability prevents accurate estimation of MF velocity and precludes conversion of spatial positions to developmental times.

\textbf{\\
}

\includegraphics[width=2.33333in,height=3.84722in]{media/image15.png}

(A-C) Effects of \emph{EGFR\textsuperscript{ts}} on (A) Pnt-GFP, (B) Yan, and (C) Pnt-to-Yan ratio dynamics in progenitor cells. Lines are moving averages across 250 sequential cells. Shaded regions are bootstrapped 95\% confidence intervals for the mean. Solid lines and grey shading denote wildtype controls. Dashed lines and red shading denote restricted EGFR signaling. Black bars denote second period of elevated Pnt-GFP expression.

\textbf{\\
}

\includegraphics[width=6.50000in,height=7.61250in]{media/image16.png}

(A) Computational pipeline for automated annotation of clonal subpopulations. (I) Example image of progenitor cells in an eye disc containing \emph{yan} null clones. (II) Watershed segmentation of a distance-transformed foreground mask of the DAPI fluorescence channel yields contours surrounding each cell nucleus. (III) Histogram of measured Ubi-mRFPnls levels within the image. Colors denote genotypes assigned by a k-means classifier. Additional clusters were manually added and merged where necessary in order to obtain reasonable thresholds between genotypes. (IV) Completed annotation of the example image. Segment colors reflect the final genotype assigned to each cell segment. Yellow, cyan, and magenta correspond to 0x, 1x, and 2x \emph{yan} genotypes, respectively. (V) Comparison with \emph{yan} genotype labels manually assigned to \textasciitilde{}2500 cells in four eye discs. Numbers denote the number of cells within each bin and the color scale denotes the overlap between bins. Overall misclassification rate is \textasciitilde{}5\%, with most errors occurring between cells with one or two copies of \emph{yan}. (B) Procedure for selection, aggregation, and comparison of clonal subpopulations. Colors retain their meaning from panel A. (I) Cell segments bordering each clone are excluded from the analysis. (II) Analysis is limited to cells coinciding with the two peaks of Pnt-GFP immediately posterior to the morphogenetic furrow. (III) Cell measurements are aggregated across multiple layers of multiple eye discs yielding statistical comparisons between clonal genotypes.

\textbf{\\
}

\includegraphics[width=6.17083in,height=9.00000in]{media/image17.png}

(A) Comparison of measured Ubi-mRFPnls and Pnt-GFP levels between annotated clone genotypes in bleed-through control clones. Measured Pnt-GFP levels are correlated with measured Ubi-mRFPnls levels despite all cells containing the same number of copies of the \emph{yan} gene. p-values are derived from Mann-Whitney \emph{U} tests.

(B) Procedure for systematic correction of RFP/GFP bleed-through applied to an example image of an eye disc containing \emph{yan} clones. (I) Background pixels are obtained by dilating a binary foreground mask until no image features remain visible. Left panel shows cell contours (foreground) superimposed on the DAPI channel of an example disc. Right panel shows dilated foreground (hashed area) superimposed on the RFP and GFP channels. (II) Background pixels are resampled such that the resultant distribution of measured RFP levels is approximately uniform. (III) A generalized linear model with gamma-distributed residuals and an identity link function is fit to the intensities of the resampled pixels' RFP and GFP channels. Data are binned for ease of visualization, red line reflects model fit. (IV) Correction of measured GFP levels (black markers) for all cells in the example image. Left and right panels show original and corrected measurements of Pnt-GFP level. The model (red line) clearly tracks the baseline trend in the original measurements, despite only being fit to the image background.

(C) Comparison of corrected Pnt-GFP levels between annotated clone genotypes in bleed-through control clones. Corrected Pnt-GFP levels are statistically indistinguishable between annotated genotypes (\emph{p\textgreater{}0.5}, Mann-Whitney \emph{U} tests).

(D) Comparison of corrected Pnt-GFP levels between annotated clone genotypes in \emph{yan} null clones. Pnt-GFP levels are higher in progenitor cells with one or two copies of \emph{yan} than in cells with no copies (\emph{p\textless{}0.001}, Mann-Whitney \emph{U} tests).
