\textbf{ABSTRACT}

Cells must reliably respond to changes in transcription factor levels in order to execute cell state transitions in the correct time and place. These transitions are typically thought to be triggered by changes in the absolute nuclear concentrations of relevant transcription factors. We have identified a developmental context in which cell fate transitions depend on changes in the relative concentrations of two transcription factors. Here, we quantify the \emph{in vivo} expression dynamics of Yan and Pointed, two essential E-twenty-six (ETS) proteins that regulate transcription during eye development in \emph{Drosophila}. These two factors exert opposing influences; one impedes transcription of gene targets required for differentiation while the other promotes it. We show that both proteins are transiently co-expressed in eye progenitor cells and also during photoreceptor specification. To decide whether to undergo state transitions, cells respond to the ratio of the two protein concentrations rather than to changes in the absolute abundance of either transcription factor. We show that a simple model based on the statistical physics of protein-DNA binding illustrates how this ratiometric sensing of transcription factor concentrations could occur. Gene dosage experiments reveal that progenitor cells stabilize the ratio against fluctuations in the absolute concentration of either protein. We further show that signaling inputs via the Notch and Receptor Tyrosine Kinase (RTK) pathways set the ratio in progenitor cells, priming them for either transit to differentiation or for continued multipotency. A sustained change in the ratio accompanies the transit to differentiation This novel mechanism allows for distributed control of developmental transitions by multiple transcription factors, making the system robust to fluctuating genetic or environmental conditions.

\textbf{INTRODUCTION}

Organismal development proceeds through a sequence of transitions that yield increasingly specific cellular states. Developmental programs employ a variety of strategies to coordinate transitions, ultimately ensuring that cells adopt the correct state in space and time. Elucidating these decision mechanisms is crucial to understanding the processes that guide development, as well as the diseases that arise when they fail.

How cells dynamically integrate the activities of one or more transcription factors to reliably execute state transitions is a long-standing question. Transcription factors initiate and enforce transitions by differentially regulating the expression of specific target genes (Zheng and Flavell 1997; Ducy et al. 1997; McGhee et al. 2009). Coordination among transcription factors can thus broaden the spectrum of possible cell responses to developmental cues (O'Riordan and Grosschedl 1999; Evans et al. 2003). Positive and negative feedback can create bi-stable patterns of mutually exclusive transcription factor expression or activity that render state transitions irreversible (Melen et al. 2005; Kueh et al. 2013; Yao et al. 2008; Park, Ahrends, and Teruel 2012). Cells can also regulate state transitions by differentially partitioning transcription factors between daughter cells (Wolff et al. 2018). In all of these models, mRNA and protein expression are believed to depend upon the absolute concentrations of the transcription factors involved (Tontonoz, Hu, and Spiegelman 1994; Laslo et al. 2006; Raj et al. 2010; Niwa, Miyazaki, and Smith 2000).

Studies of intercellular signaling suggest cells are capable of sensing relative levels of signaling molecules. For example, the TGF-β pathway elicits a cell response following changes in input signaling relative to the preceding background (Frick et al. 2017). Fold-change, rather than absolute levels of β-catenin, dictate Wnt signal transduction in eukaryotic cells (Goentoro and Kirschner 2009). Likewise, aggregation of the social amoeba \emph{Dyctiostellium} depends on fold-change detection of extracellular cAMP concentrations (Kamino et al. 2017). Relative measurements can also involve multiple molecular components. In the BMP pathway, cells interpret multi-ligand inputs based on the relative levels of each ligand pair, with additive, differential, and ratiometric response types arising directly from the relative competition between different ligand-receptor pairs (Antebi et al. 2017). In yeast, pheromone response is insensitive to the absolute abundance of the G‐protein‐coupled receptor Ste2. Instead, cells respond to the fractional occupancy of the signal receptor by forming a ratiometric sensor between Ste2 and its regulatory inhibitor Sst2 (Bush et al. 2016). Topological features of molecular interaction networks, such as feed-forward loops, can also sense relative changes in molecule abundance (Goentoro et al. 2009; Adler and Alon 2018). Combining such circuits with precise coordination of signaling inputs would yield a molecular decision mechanism that is robust to fluctuations in the abundance of participating molecules. Notably, while relevant transcription factors are precisely controlled during development (Doe 2017; Erclik et al. 2017), it remains unknown whether cells sense the relative concentrations or activities of different transcription factors when executing cell state transitions.

Two ETS-domain transcription factors, the activator Pointed (Pnt) and the repressor Yan (also known as Anterior open, Aop), are essential regulators of cell fate transitions at numerous stages of \emph{Drosophila} development (Gabay et al. 1996; Halfon et al. 2000; Morimoto et al. 1996; Xu et al. 2000; Flores et al. 2000). Consistent with their opposing regulatory effects, genetic studies have shown that Pnt and Yan can act antagonistically in guiding numerous cell fate transitions (Brunner et al. 1994; O'Neill et al. 1994; Gabay et al. 1996; Halfon et al. 2000). The \emph{pnt} locus encodes two protein isoforms, PntP1 and PntP2, that differ in their N-terminal transactivation domains but share the same DNA-binding domain (Fig. 1A) (Klämbt 1993; Scholz et al. 1993). Specifically, PntP1 is constitutively active whereas PntP2 requires phosphorylation via the RTK signaling pathway to become a potent activator (O'Neill et al. 1994; Brunner et al. 1994). Because both Pnt isoforms and Yan bind to a common ETS-binding DNA sequence motif GGA(A/T) (Wei et al. 2010), competition for occupancy and regulation of common target genes must be precisely orchestrated (O'Neill et al. 1994; Xu et al. 2000; Webber, Zhang, Cote, et al. 2013; Webber, Zhang, Mitchell-Dick, et al. 2013; Halfon et al. 2000; Flores et al. 2000; Boisclair Lachance et al. 2018; Webber et al. 2018).

Pnt and Yan display mutually exclusive expression patterns in several developing tissues (Boisclair Lachance et al. 2014). The embryonic ventral ectoderm provides a classic example in which cells unambiguously reside in one of two stable states (Melen et al. 2005; Gabay et al. 1996). These and similar observations inspired a bistable switch model of cell fate specification in which the multipotent state is defined by high absolute Yan levels and the differentiated state is defined by high absolute Pnt levels (Graham et al. 2010). The model posits that RTK signaling triggers a transition from target gene repression to activation by degrading Yan and activating PntP2 (Brunner et al. 1994; Rebay and Rubin 1995). A positive feedback loop in which transient phosphorylation of PntP2 activates expression of \emph{pntP1} is thought to stabilize the transition by enabling prolonged signaling-independent stimulation of target genes (Shwartz et al. 2013). Sustained PntP1 expression in cells devoid of Yan thereby recapitulates a complementary expression pattern under the control of RTK signaling (Boisclair Lachance et al. 2014). However, Pnt and Yan are also co-expressed during cell fate commitment in several developmental contexts (Boisclair Lachance et al. 2014). For example, the two proteins are co-expressed in posterior follicle cells of the early egg chamber and throughout the embryonic mesoderm (Boisclair Lachance et al. 2014), where they are required for specification of cell fates (Morimoto et al. 1996; Halfon et al. 2000). Co-expression also occurs in the larval eye (Boisclair Lachance et al. 2014), despite the repeated use of RTK signaling to designate cell fates (Freeman 1996). Eye development thus prompts exploration of how state transitions are resolved from concurrent Pnt and Yan expression in response to signaling cues.

Eye development is divided into two distinct phases, growth and differentiation. In the first phase, multipotent progenitor cells in the eye-field asynchronously proliferate from the earliest larval stage until the third instar stage of larval life (Wolff and Ready 1993). The differentiation phase of eye development begins in the early third instar larva, when cells situated at the posterior margin of the eye disc start to differentiate into photoreceptor (R) cells, followed by progressively more anterior cells (Fig. 1B). This wave of differentiation is initiated and coordinated by a morphogenetic furrow (MF), which traverses the eye disc from posterior to anterior for the remainder of the third instar stage up to the early pupal stage (Voas and Rebay 2004). Progenitor cells located immediately anterior to the MF arrest in G1 of the cell cycle and express a transcription factor called Atonal. Refinement of Atonal expression within this field establishes the differentiation program by specifying individual R8-type photoreceptors in a periodic pattern immediately posterior and parallel to the MF (Jarman et al. 1994; T. Zhang et al. 2006). Each R8 cell then locally secretes an RTK ligand that induces R8's multipotent neighbors to differentiate into other photoreceptor types (Fig. 1C) (Freeman 1996; Voas and Rebay 2004). Transitions sequentially occur approximately every two hours to form R2/R5, R3/R4, R1/R6, and R7 photoreceptors (Wolff and Ready 1993). Many cells remain as multipotent progenitors even after all R cell fates have been adopted. These cells will adopt other fates at later stages of eye development, with any surplus eliminated by apoptosis (Wolff and Ready 1991).

R cell fate specification violates three central tenets of the bistable switch model. First, Pnt and Yan do not exhibit a mutually exclusive expression pattern during eye development. The two proteins are extensively co-expressed in both progenitor and differentiating cells within and posterior to the MF (Boisclair Lachance et al. 2014). Second, transitioning cells do not originate in a stable high Yan state. We recently quantified Yan protein dynamics during larval and early pupal eye development (Peláez et al. 2015). In progenitor cells, Yan displays pulsatile dynamics in which protein levels rapidly increase as the MF passes and then gradually decay back to low initial values. Third, transitioning cells do not adopt a stable high Pnt state. Visual inspection of eye discs carrying a fluorescent reporter for Pnt expression suggest that Pnt levels decay on a timescale comparable to Yan in transitioning cells (Boisclair Lachance et al. 2014). Despite their similar expression patterns, the two proteins still exhibit antagonistic effects on R cell fate determination. Pnt stimulates progenitor cells to transit to an R cell fate while Yan inhibits these transitions (O'Neill et al. 1994; Rebay and Rubin 1995).

Here, we explore this apparent paradox to understand how two coexpressed transcription factors with comparable DNA binding specificity but opposing transcriptional functions elicit precisely controlled cell state transitions in the developing eye. We find that cell states are regulated by the relative abundance of these two proteins, rather than by their absolute concentrations. Progenitor cells dynamically maintain an approximately constant ratio of Pnt-to-Yan protein despite their absolute concentrations varying over time. Cells that transition to R cell fates rapidly increase their Pnt-to-Yan ratio, which remains elevated over time. We show that a ratio control strategy buffers this ratio against variable abundance of either protein. We also find that the signaling inputs of the Yan-Pnt network regulate the dynamics of the transcription factor ratio. Although Notch and Ras signals can both promote and inhibit differentiation in the developing eye (Fortini et al. 1993; Freeman 1996), we find that Notch signaling predominantly decreases the Pnt-to-Yan ratio in progenitor cells while Ras signaling mainly increases the ratio. Notch and Ras signals thus inhibit and promote cell state transitions in the eye by dynamically tuning the ratio of the two transcription factors. We conclude that progenitor cells interpret an increase in the abundance of Pnt relative to Yan as a cue to differentiate.

\textbf{RESULTS}

Although Pnt expression has been qualitatively studied in the eye (Boisclair Lachance et al. 2014), its dynamics have not been quantified. Therefore, we took advantage of a fully functional genomic transgene in which both \emph{pntP1} and \emph{pntP2} are C-terminally tagged with GFP (Boisclair Lachance et al. 2014). As previously reported (Boisclair Lachance et al. 2014), \emph{pnt-gfp} rescues \emph{pnt} null mutants to viable, fertile adults with wild type external morphology (Fig. S1A,B). As described by Boisclair Lachance et al. (2014), qualitative examination of Pnt-GFP expression in 100h eye-antennal imaginal discs revealed a region of very low expression in cells anterior to the MF, followed by strong expression in two parallel stripes of cells immediately posterior to the MF (Fig. 1D, regions 1 and 2).

\textbf{The Pnt-to-Yan ratio varies between cells in different states }

We used Histone-RFP fluorescence from a His2Av-mRFP transgene to label all eye cell nuclei for automated segmentation following direct fluorescence microscopy of fixed specimens (Peláez et al. 2015; Peláez 2016). Average Pnt-GFP fluorescence levels and exact 3D positions were then calculated for all nuclei in each developing eye disc. Pnt-GFP fluorescence levels were normalized to Histone-RFP, which provided some control over measurement noise and nuclear constriction that occurs at the MF (Peláez et al. 2015; Peláez 2016). We mapped each cell's position along the anterior-posterior coordinate of the eye disc to a point in developmental time. This linear approximation is sufficiently accurate because the MF moves across the eye field with approximately constant velocity, forming one column of R8 cells every two hours (Basler and Hafen 1989; Campos-Ortega and Hofbauer 1977). The distance between a cell and the MF is therefore proportional to the time elapsed since the MF passed. We also manually assigned a state value to each cell. Cell state classification is possible because nuclei can be unambiguously identified without cell-specific markers by their morphology, apical-basal position, and relative distance to the furrow (Ready, Hanson, and Benzer 1976; Tomlinson 1985; Tomlinson and Ready 1987; Wolff and Ready 1993; Peláez et al. 2015; Peláez 2016). Combined, these data allowed us to infer a macroscopic view of cell state transition dynamics from the spatial arrangement of cells relative to each other and the MF. Although our approach cannot measure the developmental progression of an individual cell, it provides a dynamic view of thousands of cells across a developing eye. From this information, average cell behaviors can be reconstructed and modeled.

Progenitor cells anterior to the MF expressed a basal level of Pnt-GFP, but expression dramatically increased in cells immediately anterior to the MF (Fig. 1E). This was followed by two successive pulses of Pnt-GFP expression, marked by peaks where protein expression reached maximal amplitude. The pulses matched the visual stripes seen in regions 1 and 2 (Fig. 1D). Thereafter, Pnt-GFP decayed to a low basal level. The two pulses of Pnt-GFP in progenitor cells coincided with the two periods of transition to R cell states (Figs. 1F,G and S1C). Transitions to R8, R2/R5, and R3/R4 states occurred during the first pulse, while transitions to R1/R6, and R7 states occurred during the second pulse.

Based on prior description of the distinct temporal expression patterns of isoform-specific \emph{pnt} transcriptional reporters (Shwartz et al. 2013), we suspected that each pulse of Pnt-GFP corresponded to the induction of either PntP1 or PntP2. Using flies that carried the \emph{pnt-gfp} transgene and either a \emph{pntP1-} or \emph{pntP2}-specific reporter, we found that region 1 overlapped with the domain of strongest \emph{pntP1} reporter expression, and region 2 corresponded to the domain of strongest \emph{pntP2} reporter expression (Fig. 1H,I). Low levels of \emph{pntP2} reporter expression were detected in region 1 and low levels of \emph{pntP1} reporter were detected in region 2. Therefore, Pnt expression appears as a PntP1-PntP2 pulse sequence. The two groups of differentiating R cells predominantly expressed PntP1 or PntP2 respectively (data not shown, see Peláez 2016), suggesting that specific Pnt isoforms are used to specify distinct cell fates.

All cell state transitions coincided with a rapid increase in Pnt-GFP (Fig. 1F,G). The earliest identified R cells had, on average, 25-50\% higher levels of Pnt-GFP than progenitor cells at comparable times. Pnt-GFP then rapidly decayed in all differentiating R cells, with all but the R7 cell type exhibiting faster decay kinetics than progenitor cells. Thus, transitioning R cells did not adopt stable high Pnt levels as predicted by a bi-stable model of R cell fate specification. Rather, the measured Pnt-GFP expression dynamics were similar to those previously reported for Yan (Peláez et al. 2015). Average levels of both proteins increased as the MF passed, then decayed during cell state transitions. These similar population-wide dynamics led us to ask how cell states are resolved from the co-expression of two transcription factors with opposing transcriptional functions at the single cell level.

We explored this question by simultaneously measuring Pnt-GFP and Yan in each nucleus using an anti-Yan monoclonal antibody. Previously, we had shown that Yan dynamics measured with the antibody were almost identical to those measured by a YFP tagged version of Yan (Peláez et al. 2015), validating our approach. Pnt-GFP and Yan were induced at the same time in progenitor cells (Figs. 1E,J and S1D). Yan levels reached a maximum amplitude between the two pulses of Pnt-GFP. Yan then decayed back to a basal steady-state level, interrupted by a transient plateau during the second pulse of Pnt-GFP. Despite their alternating maxima, the overall induction and decay of Pnt-GFP and Yan were concurrent in progenitor cells.

The similar dynamics prompted us to consider whether relative levels of Pnt and Yan dictate cell state transitions in the eye. To explore this possibility, we measured the ratio of Pnt-to-Yan in each progenitor cell. Strikingly, the average Pnt-to-Yan ratio remained dynamically stable about a constant value over time (Fig. 1K). However, from 0 to 15 hours, there was considerable cell-to-cell heterogeneity in the ratio. Some cells had above-average ratios when they were expressing peak levels of Pnt-GFP, and many cells had below-average ratios when they were between the Pnt-GFP pulses. After the second pulse of Pnt-GFP expression, cells acquired a slight bias towards Yan. These are the progenitor cells that remain multipotent and are used to differentiate into other cell types later in development (Wolff and Ready 1991). As the two positive spikes in the ratio coincided with the two periods of cell state transition, we reasoned that dynamic changes in the ratio might control the state of cells and regulate their transit to differentiation.

As a first test of this idea, we quantified the levels of Pnt-GFP and Yan in cells that had undergone R cell state transitions. We focused on R2/R5 and R1/R6 cells, since they are representative of transitions of cells derived from groups 1 and 2, respectively. As previously noted, Pnt-GFP levels were elevated in both sets of ``young'' R cells as soon as we could confidently identify them (Fig. 2A,B). In contrast, Yan levels were lower in these young R cells than in progenitor cells of comparable age (Fig. 2C,D). This meant that the average ratio of Pnt-to-Yan was elevated 1.5- to 2-fold in young R cells (Fig. 2E,F). The ratio elevation was more modest in young R2/R5 cells than in young R1/R6 cells, but all increases were significant (p\textless{}0.001, KS test). The elevated ratio persisted for all times thereafter as differentiation proceeded. Analogous ratio trends were evident with the other R cell types as well (Fig. 2G and S2), suggesting that different state transitions share a common requirement for sustained change in the Pnt-to-Yan ratio relative to progenitor cells (Fig S2 C-G).

We next asked whether elevated Pnt-to-Yan ratios precede the onset of R cell state transitions. If R cells are recruited from a subpopulation of progenitors with relatively high ratios, some progenitors should exhibit ratios comparable to those of early R cells. We compared the distribution of Pnt-to-Yan ratios in young R cells versus concurrent progenitor cells (Fig. 2H,I). The extensive overlap between the two populations implies that some cells we morphologically classified as progenitors were actually transitioning to R cell fates. Additionally, ratios may have increased among a subset of progenitors that did not ultimately transition to an R cell state. Many progenitors also adopted low Pnt-to-Yan ratios during this time period that did not overlap with the early R cell population. We reasoned that these cells were not viable candidates for recruitment, suggesting that the extent of variation in the ratio among progenitors constrains their competence for differentiation.

We then asked whether variation in the Pnt-to-Yan ratio strictly coincides with R cell fate transitions. We anticipated that variability should arise within the pool of progenitors from which R cells are recruited, then subside as fates are resolved. We previously reported methods to quantify the dynamic cell-to-cell heterogeneity of Yan concentration (Peláez et al. 2015). We applied similar analysis to simultaneously quantify the heterogeneity of Pnt, Yan, and the Pnt-to-Yan ratio among both progenitors and early R cells (Fig. S3). A broad increase in variation of the ratio among progenitors coincided with the time periods in which state transitions occurred. Ratio variation was predominantly attributed to Yan and Pnt variability during the first and second groups of R cell state transitions, respectively. Heterogeneity among progenitors then decreased back to basal levels after all R cell fate transitions were complete. Similar trends were evident among transitioning R cells, but with a more rapid approach toward a consensus ratio following fate specification.

\textbf{Cooperative DNA-binding sensitizes transcriptional output to changes in Pnt-to-Yan ratio }

How could cells reprogram transcription in response to a change in the Pnt-to-Yan ratio across a wide range of absolute protein concentrations? Since both transcription factors have overlapping sequence specificity for DNA binding (Wei et al. 2010; Webber, Zhang, Cote, et al. 2013; Webber, Zhang, Mitchell-Dick, et al. 2013; Nitta et al. 2015; Xu et al. 2000; Halfon et al. 2000; Flores et al. 2000), the underlying mechanism may be a natural consequence of competition for binding sites in target genes. A simple model in which the two transcription factors compete for occupancy of common binding sites shows that if the sites are saturated, then equilibrium occupancy by either factor is more sensitive to the relative concentration of the two factors than to the absolute concentration of both species (Fig. S4A,B).

However, the situation is more complex for Pnt and Yan. While there are several well-documented target genes that contain common binding sites for Pnt and Yan (Halfon et al. 2000; Xu et al. 2000; Flores et al. 2000; Boisclair Lachance et al. 2018), Yan binds these enhancers with higher affinity than Pnt (Xu et al. 2000). Moreover, recent experiments suggest a scenario in which Yan and Pnt differentially interpret the structural syntax of cis-regulatory modules (Boisclair Lachance et al. 2018). This complex phenomenon is a consequence of cooperative recruitment between adjacent chromatin-bound Yan molecules. Yan monomers are able to polymerize via their sterile alpha motif (SAM) binding domains, enabling tightly-bound Yan monomers at strong ETS sites to stabilize the recruitment of additional Yan monomers to adjacent, weaker ETS sites or non-ETS sites (Qiao et al. 2004; Boisclair Lachance et al. 2018). These cooperative effects could conceivably bias the competition between Yan and Pnt, which led us to consider a more complex model.

A modeling framework was recently introduced in order to probe the effects of \emph{cis}-regulatory syntax on Yan binding site occupancy (Hope, Rebay, and Reinitz 2017). The model considers an ensemble of microstates, each defined by a unique configuration of vacant or Yan-bound sites. Each microstate is assigned a thermodynamic potential based on the cumulative influence of strong ETS-binding, weak non-ETS binding, and polymerization. We augmented this model by incorporating Pnt as a second transcription factor that competes for occupancy of the same binding sites (Fig. S5A,B). Using this model, we sought to characterize the sensitivity of Pnt binding site occupancy to changes in the Pnt-to-Yan ratio without neglecting cooperativity derived from \emph{cis}-regulatory syntax.

We first considered a scenario in which Yan and Pnt did not exhibit cooperativity (Fig. 3A). In the absence of stabilizing SAM-SAM interactions, the landscape of overall binding site occupancy is identical to that obtained with the simple binding model described above (Figs. 3B and S4B). Increasing the Pnt-to-Yan ratio revealed a gradual increase in Pnt occupancy for all individual binding sites (Fig. 3C). This titration contour closely resembles a Langmuir isotherm or Michaelis-Menten saturation curve.

We then introduced a stabilizing SAM-interaction for Yan (Fig. 3D). The resultant landscape of overall Pnt binding site occupancy is clearly distinguished from the simple binding model by a sharpening of the transition from Yan to Pnt dominance in occupancy (Fig. 3E). Weighting the energetic contributions of binding strength and polymerization by the statistical frequency of each microstate revealed that the transition is driven by an abrupt change in the dominant binding mechanism. Polymerization effects dominate binding site occupancy when the Pnt-to-Yan ratio is low, while binding strength dominates when the ratio is high (Fig. S5C).

Increasing the Pnt-to-Yan ratio revealed nonlinear transitions from low to high Pnt occupancy for each individual binding site (Fig. 3F). These transitions resemble Hill functional forms, indicating the emergence of sharp thresholds that delimit distinct regimes of transcriptional output. At low Pnt-to-Yan ratios, Yan is able to polymerize and occupies all binding sites. At some critical Pnt-to-Yan ratio, Pnt-bound sites intersperse Yan-bound sites such that Yan is no longer able to polymerize. Pnt then outcompetes Yan as the ratio increases further. These results recapitulate the long-standing notion that cooperative DNA-binding sensitizes transcriptional output to changes in transcription factor activity.

Assuming binding sites are saturated, then relative occupancy by Pnt and Yan is agnostic to changes in the absolute abundance of either factor, as long as the Pnt-to-Yan ratio remains constant. This mechanism, coupled with cooperativity, would enable modest changes in the Pnt-to-Yan ratio to elicit large changes in DNA binding site occupancy by either factor, and presumably large changes in mRNA synthesis given the opposing transcriptional effects of Yan and Pnt.

\textbf{Regulation stabilizes the Pnt-to-Yan ratio against varying Pnt and Yan concentrations}

Equilibrium modeling suggests cell state transitions can proceed normally amidst individual or cell-to-cell fluctuations in the absolute concentrations of Pnt and Yan. We tested this idea by varying the genetic dosage of the \emph{pnt} gene from one to four copies. Protein output in \emph{Drosophila} is proportional to the number of copies of any given gene (Lucchesi and Rawls 1973), validating our strategy. We found that the eyes of adult flies were morphologically indistinguishable across this \emph{pnt} dosage range (Fig. S1A,B). A similar lack of dosage sensitivity had been previously observed with the \emph{yan} gene (Peláez et al. 2015). Because both sets of genetic manipulations should in theory change the Pnt-to-Yan ratio, the absence of overt phenotypes suggested that either cell state transitions are not sensitive to this ratio or that there are active feedback mechanisms that drive cells back to the ideal ratio.

To distinguish between these possibilities we asked whether the ratio of Pnt-GFP to Yan protein is sensitive to the abundance of Pnt-GFP protein. We quantified Pnt-GFP levels in eye cells containing either one or two copies of the \emph{pnt-gfp} transgene in a \emph{pnt} mutant background. As expected, Pnt-GFP protein concentration varied proportionally to \emph{pnt-gfp} gene copy number (Fig. 4A,B). Interestingly, average Yan protein concentration also scaled with \emph{pnt-gfp} gene copy number (Fig. 4C,D) resulting in an essentially identical Pnt-to-Yan protein ratio (Fig. 4E). The dependence of Yan protein output on \emph{pnt-gfp} gene copy number parallels our previous finding that \emph{pnt} mutant cells had lower Yan protein concentrations (Peláez et al. 2015). We conclude that the network regulating Yan protein output compensates for variation in the abundance of Pnt to maintain a constant Pnt-to-Yan ratio.

Effective control of the Pnt-to-Yan ratio would require a similar dependence of Pnt protein output on the concentration of Yan. We assessed this prediction by quantifying Pnt-GFP protein in cells with different copy numbers of the \emph{yan} gene. Due to the embryonic lethality of \emph{yan}-null mutations, we conducted this experiment by inducing \emph{yan}-null clones in the developing eye, and using an Ubi-mRFPnls marker to identify genotypes of cells in the clones. Cells with different \emph{yan} gene dosages exhibited different Pnt-GFP protein concentration (Figs. 4F,G and S6A,B). Notably, Pnt-GFP levels were correlated with \emph{yan} gene dosage in progenitor cells immediately posterior to the MF. We quantified this effect by measuring Pnt-GFP protein concentration in cells with or without a copy of the wildtype \emph{yan} gene (Figs. 4H and SI-2D). Pnt-GFP expression was higher in cells with one or more copies of \emph{yan} than in cells with zero copies (Mann-Whitney \emph{U} test, p\textless{}0.001), suggesting that Pnt protein output is dependent on the abundance of Yan protein. Overall, these data suggest cells compensate for fluctuations in Pnt or Yan protein abundance by adjusting the level of the opposing protein. While Yan expression was not quantified in this experiment due to technical limitations, mutual compensation could account for the constant Pnt-to-Yan ratio in progenitor cells, despite large changes in Pnt and Yan concentration over time.

\textbf{Notch signaling lowers the Pnt-to-Yan ratio in progenitor cells}

Progenitor cells are dynamically stable about a constant Pnt-to-Yan ratio, but this fixed ratio changes as cells transition to an R cell state. We next asked how the ratio is set in individual cells targeted for differentiation. Notch and RTK signaling provide both transcriptional and post-transcriptional inputs to the Yan-Pnt network (Graham et al. 2010), and therefore present prime candidates to establish and modulate the Pnt-to-Yan ratio. The two pathways generally exert opposing influence on R cell state transitions. Notch signaling is required to maintain progenitor cells in a multipotent state (Fortini et al. 1993), and ensures proper patterning of the first group of R cells by constraining the proximity of adjacent R8 cells (Lubensky et al. 2011; Gavish et al. 2016). Conversely, RTK signaling is required to initiate each of the subsequent R cell state transitions (Freeman 1996). Both signaling pathways influence the Pnt-Yan network in multipotent eye cells (Brunner et al. 1994; Rebay and Rubin 1995; Rogge et al. 1995). Notch stimulates Yan expression (Rohrbaugh et al. 2002), while RTK activates PntP2 and stimulates PntP1 expression (Brunner et al. 1994; Shwartz et al. 2013) while attenuating the pulse of Yan expression (Rebay and Rubin 1995; Peláez et al. 2015). The precise influence of Notch and RTK signaling on the Pnt-to-Yan ratio are difficult to predict as the two pathways are coupled by feedback within and beyond the Pnt-Yan network (Rohrbaugh et al. 2002; Kumar et al. 2003; Voas and Rebay 2004).

We used a temperature-sensitive \emph{Notch} mutant (Cagan and Ready 1989) to measure the impact of Notch-mediated signaling on Pnt-GFP and Yan. We divided the eye field into two regions for analysis purposes (Fig. S7A, dashed yellow line). The first region starts anterior to the MF and ends in the region of R8 cell specification. The second region extends \textasciitilde{}10 columns of R8 cells posterior to the first region. These regions contain the first and second pulses of Pnt-GFP, respectively (Fig. S7A, thick black lines).

At the non-permissive temperature, progenitor cells had visibly reduced Yan levels (Fig. S7A,B), consistent with previous reports of Yan expression's dependence on Notch signaling in the developing eye disc (Rohrbaugh et al. 2002). Pnt-GFP levels were also reduced in region 1, but appeared close to normal at later times (Fig. S7A,B). Attempts to study ratio dynamics in mutant eye discs were challenging. Notch is essential for proper patterning of R8 cells, so the mutant eye discs had distorted spacing of R8 cells at the non-permissive temperature (Fig. S7C). The irregularity of the intervals between adjacent columns of R8 cells precluded the conversion of spatial position along the anterior-posterior axis to developmental time. As an alternative to our standard quantitative analysis, we visualized the effects of Notch by mapping the pixel-wise difference between Pnt-GFP and Yan to a diverging color scale (Fig. 5A,B). Direct visualization of the ratio yielded a very similar view of differential Pnt-GFP/Yan expression, but was prone to computational errors imparted by zero-valued pixels. This qualitative analysis was limited to optical sections specifically spanning the progenitor cells.

At the non-permissive temperature, the \emph{Notch} mutant showed a consistently higher Pnt-to-Yan ratio in progenitor cells in region 2 (Fig. 5A). This observation suggests that Notch signaling maintains a low Pnt-to-Yan ratio in these undifferentiated cells. The known role of Notch in maintaining multipotency in region 2 (Fortini et al. 1993) is consistent with our hypothesis that cell state transitions are mediated by the transcription factor ratio.

The Notch mutant revealed more complex behavior in region 1, where the first group of R cell transitions occur. In this region Notch is required for patterning of the R8 lattice (Lubensky et al. 2011). At the permissive temperature, there was a periodic pattern in the Pnt-to-Yan ratio of progenitor cells (Fig. 5B). Clusters of cells with higher ratio alternated with clusters with lower ratio. We quantified the periodicity of this pattern by evaluating the similarity of ratios between cells as a function of their separation distance (Fig. 5C). We detected periodic spatial patterns with a constant period of oscillation that was approximately equivalent to the length scale separating adjacent R8 cells (Fig. 5D,E). Since young R8 cells have elevated Pnt-to-Yan ratios, we infer that periodic clusters of high-ratio progenitor cells give rise to the R8, R2/R5, and R3/R4 cells. At the non-permissive temperature, the Pnt-GFP/Yan pattern was strongly impaired (Fig. 5B). The ratio was more uniform along the dorso-ventral axis, and while there were modestly detectable oscillations in the ratio, their period was not stable (Fig. 5F,G).

These results are consistent with the consensus understanding that Notch signaling serves dual roles in R8 fate determination. Initially, Notch pushes clusters of progenitor cells towards an R8 cell state. Later, Notch restricts differentiation to ensure only one cell per cluster adopts an R8 state (Baker and Yu 1997; Li and Baker 2001; Lubensky et al. 2011). If Notch signaling is inhibited, very few high Pnt-to-Yan ratio clusters and R8 cells are formed because the first step is blocked (Baker and Yu 1997). If cell state transitions are coupled to the Pnt-to-Yan ratio, then the \emph{Notch} mutant would not be expected to have high-ratio clusters in region 1. This is precisely what we observed.

\textbf{Ras signaling elevates the Pnt-to-Yan ratio in progenitor cells}

RTK signals received by progenitor cells trigger their transition to R cell states (Freeman 1996). We quantitatively probed the effect of RTK signaling on Pnt-GFP dynamics and the Pnt-to-Yan ratio by using a temperature-sensitive \emph{EGFR\textsuperscript{ts}} allele that restricts RTK signaling (Kumar et al. 1998). At high temperatures, the mutant blocks RTK signal transduction, which triggers cell death and allows only R8 neuron patterning. However, animals raised at intermediate temperatures achieve normal recruitment of R8, R2/R5, and R3/R4 neurons, but fail to recruit most R1/R6 and R7 cells (Peláez et al. 2015).

At intermediate temperatures, the entire second pulse of Pnt-GFP expression disappeared upon restriction of EGFR activity (Fig. S8A). Because the second pulse can be predominantly ascribed to PntP2 expression (Fig. 1I), this result conflicts with a report that PntP2 expression is not dependent upon RTK signaling (Shwartz et al. 2013). At intermediate signaling, Yan levels were also reduced, resulting in Pnt-to-Yan ratios that were indistinguishable from wildtype during the second wave of state transitions (Fig. S8B,C).

We next asked whether RTK signaling is sufficient to induce an increase in the Pnt-to-Yan ratio of progenitor cells by expressing a constitutively-active form of Ras (Fortini, Simon, and Rubin 1992; Simon et al. 1991). This construct uses a \emph{sev} promoter to drive Ras expression, limiting its effects to the second region of Pnt-GFP expression. Constitutive Ras activity dramatically increased the amplitude and duration of the second pulse of Pnt-GFP expression in progenitor cells (Fig. 6A) but did not significantly alter Yan expression dynamics (Fig. 6B). This yielded a sustained increase in Pnt-to-Yan ratio in progenitors during the second wave of state transitions (Fig. 6C), as well as the ectopic differentiation of R cells. As previously reported (Fortini, Simon, and Rubin 1992), supernumerary cells included extra R7 cells, as well as additional R cells whose identities were not discerned due to their aberrant positioning.

We sought to determine whether these ectopic R cells emerged from a pool of progenitors with abnormally high Pnt-to-Yan ratios. Focusing our analysis on progenitor cells concurrent with the ectopic induction of R cells revealed that their ratios were higher than those of wildtype progenitor cells at comparable times (Fig. 6D,E grey boxes). Nearly all young supernumerary R cells had ratios that were within the range of mutant progenitor cells, and were above the range observed in wildtype cells (Fig. 6D,E blue and purple markers). These observations suggest that abnormally high Pnt-to-Yan ratios in progenitor cells accompany ectopic R cell state transitions.

\textbf{DISCUSSION}

Successful cell state transitions usually require changes in mRNA and protein expression, which are often dictated by transcription factors. It is widely believed that expression depends on the absolute concentration of these factors (Spitz and Furlong 2012). We have identified a scenario in which cells respond to the ratio in abundance of two transcription factors rather than to the absolute concentration of either protein. This novel mechanism is made possible by several characteristics of the system. First, target genes of these factors are induced when cells transit to differentiated states, and expression of these targets is necessary for differentiation (Xu et al. 2000; Nagaraj, Canon, and Banerjee 2002). Second, both factors bind to the same DNA sites in their target genes. Third, they exert opposing effects on transcription.

A theoretical analysis reveals that, under these conditions, relative site occupancy by either factor determines whether or not a target gene is transcribed. When binding sites are saturated, the probability a site is occupied by one of the factors is controlled by the ratio of factor concentrations and not by the absolute concentration of the factors. This mechanism allows the transcription factors to have pulsatile expression dynamics and still consistently regulate transcription. Regulation of genes through pulsatile dynamics of competing transcription factors with opposing effects has been reported in yeast (Lin et al. 2015). In the developing eye, however, this competition relies on the ratio of the two factors to differentially regulate genes. Perhaps the pulsatile dynamics of Pnt and Yan allow R cell state transitions to be restricted to a specific period of developmental time.

The proposed ratiometric sensing mechanism is well suited when either or both transcription factors bind their target sites in a manner stabilized by cooperative interactions. Like its human ortholog TEL1 (Kim et al. 2001), DNA-bound Yan monomers enhance recruitment of Yan to adjacent binding sites through stabilizing SAM-SAM polymerization (Zhang et al. 2010). We show theoretically that such cooperativity could generate threshold-like behavior and cause ultrasensitive switching in site occupancy between Yan and Pnt. In this model, the switching is agnostic to the absolute abundance of either transcription factor as long as their relative ratio can be precisely controlled. This mechanism would enable state transitions to proceed despite variation in protein concentrations driven by fluctuations in metabolism, cell volume, expression noise, environmental conditions, genetic polymorphisms, or gene copy number.

The average Pnt-to-Yan ratio is dynamically stable about an approximately constant value within each cellular state, and only exhibits transient fluctuations when cells switch states. These dynamics reflect the capacity of the system to coordinate the relative expression of the two transcription factors. We have found that the abundance of each transcription factor depends upon the expression of the other factor. Dependencies of this type are often depicted as positive or negative regulation in cartoons of qualitative regulatory interactions (Fig. 7A). We believe that as biology becomes increasingly quantitative it will be more fruitful to emphasize an empirical description of system dynamics based on control theory (Fig. 7B).

From a control perspective, a system of cellular components monitors the relative abundance of Pnt and Yan and takes corrective action when the ratio deviates from a specified reference value. The particular components responsible for implementing control may remain unspecified. This perspective eschews molecular events in favor of minimizing complexity, but preserves the salient features of a detailed molecular mechanism and can enable quantitative predictions.

Fluctuations in the absolute abundance of one factor are mitigated by compensatory adjustment of the other. Notch or RTK activity could modulate Pnt or Yan protein levels to transiently perturb the Pnt-to-Yan ratio (Fig. 7B, dashed black arrow). These signals could permanently set the ratio by adjusting the reference value (Fig. 7B, dashed red arrow). We advocate this control theoretic perspective because it more accurately conveys the fundamental strategy underlying system behavior. Furthermore, accurate model predictions would only require the evaluation of a small number of parameters that characterize Pnt-to-Yan ratio dynamics, obviating the need for experimental measurement of reaction rates during R cell specification in \emph{Drosophila}.

We have shown that dynamic changes in the Pnt-to-Yan ratio are coupled to cell state. Our favored interpretation is that the ratio determines which state a cell is in. Direct testing of this hypothesis remains difficult because the observed ratio control strategy precludes gene-level manipulation of the ratio. Instead, we acknowledge that our observations are correlative. We cannot discard the possibility that Notch and RTK signaling regulate state transitions in a manner that is not only mediated by the Pnt-to-Yan ratio but by other mechanisms as well. We emphasize, however, that ratio control enforces stability when cells are in one state, which implicates the ratio as an active cell state determinant. Moreover, qualitatively different approaches to experimentally manipulate the ratio affected cell state transitions in a consistent manner. Both Notch inhibition and Ras activation increase the ratio and cause abnormal R cell state transitions (Yang and Baker 2006; Fortini, Simon, and Rubin 1992). Single-cell dynamical measurements of Yan and specific Pnt isoforms during isoform-specific perturbations may ultimately prove necessary to determine definitively whether ratios directly mediate transitions. The regulatory mechanism described here provides insight into how the relative dynamics of competing transcription factors can be used to pattern complex epithelia, and may also aid design of synthetic regulatory systems based on ratiometric sensing.

\protect\hypertarget{_gjdgxs}{}{}\textbf{METHODS}

\textbf{Genetics:}

The recombineered~\emph{pnt-gfp} BAC transgene inserted into the VK00037 landing site was previously described in Boisclair-Lachance et al. (2014). Alleles \emph{pnt\textsuperscript{Δ88}} (O'Neill et al. 1994) and \emph{pnt\textsuperscript{2}} (Bloomington Stock 2222) were used to render the endogenous \emph{pnt} gene null in the presence of \emph{pnt-gfp}. A single copy of \emph{pnt-gfp} rescued \emph{pnt\textsuperscript{Δ88}}/\emph{pnt\textsuperscript{2}} to full viability and fertility (Fig. S1A). Cell nuclei of developing eye-antennal discs were marked by recombining \emph{H2Av-mRFP} (Bloomington stock 23651) with \emph{pnt\textsuperscript{2}}. Experiments measuring wild type dynamics of \emph{Pnt-GFP} were done by dissecting eye discs from white prepupae carrying \emph{w\textsuperscript{1118} ; pnt-gfp} \emph{/ pnt-gfp; pnt\textsuperscript{Δ88}/pnt\textsuperscript{2}, H2Av-mRFP.} \emph{Pnt} isoform-specific expression was detected using enhancer traps \emph{HS20} (gift from B. Shilo) and \emph{pnt\textsuperscript{1277}} (Bloomington stock 837), which report \emph{PntP1} and \emph{PntP2} transcription respectively by expressing LacZ (Scholz et al. 1993). \emph{pnt-gfp} and \emph{pnt} isoform-specific expression were compared in white prepupae carrying \emph{w\textsuperscript{1118} ; pnt-gfp} \emph{/ pnt-gfp; HS20/+ and w\textsuperscript{1118} ; pnt-gfp} \emph{/ pnt-gfp; pnt\textsuperscript{1277}/pnt\textsuperscript{1277}}. \emph{Pnt} gene dosage experiments were done using \emph{w\textsuperscript{1118 }}; \emph{pnt-gfp} \emph{/+ ; pnt\textsuperscript{Δ88}/pnt\textsuperscript{2}} (1x \emph{pnt}) and \emph{w\textsuperscript{1118} ; pnt-gfp} \emph{/ pnt-gfp; pntΔ88/pnt\textsuperscript{2}} (2x \emph{pnt}). Notch activity was conditionally reduced using the \emph{N\textsuperscript{ts1}} temperature sensitive allele (Shellenbarger and Mohler 1975). \emph{N\textsuperscript{ts1}/N\textsuperscript{ts1} ; pnt-gfp} \emph{/+} animals were raised at the permissive temperature (18°C) and shifted to the restrictive temperature (28.5°C) as third instar larvae for 24h. Animals exposed to the restrictive temperature that were transferred back to the permissive temperature had roughened eye phenotypes and a notched wing phenotype as adults, consistent with effective inhibition of Notch activity. Control larvae of the same genotype were grown at the permissive temperature until dissection. Both control and heat-treated larvae were sexed and only \emph{N} hemizygote males carrying \emph{N\textsuperscript{ts1}/Y ; pnt-gfp} \emph{/+} were dissected as white prepupae\emph{.} EGFR activity was conditionally reduced by placing~the null allele \emph{egfr\textsuperscript{f24}}~- also known as \emph{egfr\textsuperscript{CO}} (Clifford and Schupbach 1989)~\emph{in trans}~to the thermo-sensitive allele \emph{egfr\textsuperscript{tsla}}~(Kumar et al. 1998), as previously described (Peláez et al. 2015). The genotype was \emph{w\textsuperscript{1118 }}; \emph{egfr\textsuperscript{tsla}, pnt-gfp} \emph{/egfr\textsuperscript{24}, pnt-gfp}. Ras activation was achieved using a transgene expressing a~\emph{Ras1\textsuperscript{V12}}~mutant and driven by a 3x\emph{sev} enhancer and promoter (Fortini, Simon, and Rubin 1992) as previously described (Peláez et al. 2015). \emph{Pnt-gfp} in the Ras mutant background was measured using \emph{w\textsuperscript{1118 }}; \emph{pnt-gfp, Sev\textgreater{}Ras\textsuperscript{v12}/ pnt-gfp; pnt\textsuperscript{2}, H2Av-mRFP/+}. Controls animals carried \emph{w\textsuperscript{1118 }}; \emph{pnt-gfp} \emph{/ pnt-gfp; pnt\textsuperscript{2}, H2Av-mRFP/ +. Yan} mutant eye clones were generated using the \emph{yan\textsuperscript{833}} null allele (Webber, Zhang, Mitchell-Dick, et al. 2013)\emph{, ey\textgreater{}FLP} and the FRT40 crossover point\emph{. Pnt\textsuperscript{+}} tissue was labeled using the clonal marker \emph{Ubi\textgreater{}mRFP\textsubscript{nls}} (Bloomington Stock 34500). Developing eyes were dissected from white prepupae carrying \emph{w, ey\textgreater{}FLP ; pnt-gfp, yan\textsuperscript{833}, FRT40A/ pnt-gfp, Ubi\textgreater{}mRFP\textsubscript{nls} , FRT40A.} Control discs to measure the GFP-mRFP fluorophore bleed-through were obtained from flies carrying \emph{w, ey\textgreater{}FLP ; pnt-gfp, Ubi\textgreater{}mRFP\textsubscript{nls} , FRT40A/ pnt-gfp, FRT40A or w, ey\textgreater{}FLP ; pnt-gfp, Ubi\textgreater{}mRFP\textsubscript{nls} , FRT40A/CyO.}

\textbf{Immunohistochemistry}

Unless otherwise noted, Pnt-GFP and Yan were measured in developing animals raised at 21°C, selected as white prepupae, and subsequently aged in humid chambers for 5-10 h. Eye-antennal discs were dissected in PBS, and fixed in 4\% (w/v) paraformaldehyde/PBS for \textasciitilde{}45 min. Endogenous Yan protein was detected with the mouse monoclonal anti-Yan antibody 8B12 (Developmental Studies Hybridoma Bank, 1:200 dilution) and the secondary goat anti-mouse Pacific Blue antibody (Life Technologies, 1:200 dilution). Expression from the \emph{HS20} and \emph{pnt\textsuperscript{1277}} enhancer traps was detected using mouse anti-β-galactosidase 40-1a (Developmental Studies Hybridoma Bank, 1:50 dilution). H2Av-mRFP as a nuclear marker was used as previously described in Peláez et al. (2015). Discs were incubated in 1:1 (v/v) PBS:VectaShield (Vector Laboratories) for 45 min, then in 100\% VectaShield for an additional 45 min before mounting.

For experiments using either \emph{yan} mutant clones, \emph{N\textsuperscript{ts}}, and \emph{EGFR\textsuperscript{ts}} alleles, nuclei were stained with 4′,6-diamidino-2-phenylindole (DAPI) as a nuclear marker. Samples were fixed in 4\% paraformaldehyde, rinsed with PBS-Tween 0.5\%, and permeabilized with PBS-Triton X-100 0.1\% for 20 minutes at room temperature. Permeabilization was important to allow DAPI penetration without perturbing the fluorescence of the Pnt-GFP protein. After permeabilization, eye discs were incubated in a blocking solution containing PBS-Tween 0.1\% and 1\% normal goat serum for 30 minutes at room temperature. Primary and secondary antibodies were incubated each for 2 hours at room temperature. Antibodies used with DAPI were: mouse anti-Yan 8B12 (DHSB, 1/500) and goat anti-mouse Cy3 (1/2000, Jackson Immunoresearch). Discs were mounted in 0.5\% n-propyl-gallate, 0.1M Tris pH 8.0 and 90\% glycerol.

Samples were kept in the dark at -20°C and imaged no later than 18-24~hr after fixation. In all cases, 1024 x 1024 16-bit images were captured using either a Zeiss LSM880 or a Leica SP5 confocal microscope equipped with 40X oil objectives. During imaging, discs were oriented with the equator parallel to the x-axis of the image. Optical slices were set at 0.8µm slices (45-60 optical slices) with an additional digital zoom of 1.2-1.4 to completely image eye discs from basal to apical surfaces. Images recorded a region of at least 6 rows of ommatidia on each side or the dorsal-ventral eye disc equator. All discs for a given condition were fixed, mounted, and imaged in parallel to reduce measurement error.

\textbf{Quantification of expression levels}

Expression dynamics were inferred from confocal image stacks using an updated version of an existing segmentation and annotation pipeline (Peláez et al. 2015). The new pipeline includes \emph{FlyEye Silhouette}; an open-source package for macOS that integrates our image segmentation algorithm with a GUI for cell type annotation. Subsequent analysis and visualization procedures were implemented in Python.

In all cases, cell segmentation was performed using either H2Av-mRFP (Figs. 1,2, S1, S2, S3, and 6) or DAPI (Figs. 4-5, S8, SI1, SI-2) signals as a reference channel for identification of cell nuclei boundaries. Each layer of the reference channel was segmented independently. A single contour containing each unique cell was manually selected and assigned a cell type using a custom graphic user interface. DAPI-stained discs were segmented using a separate script based on the watershed algorithm in order to mitigate the effect of bright spots caused by DAPI accumulation in nucleoli. Further care was taken to avoid annotating contours containing such nucleoli. For each annotated cell contour, expression measurements were obtained by normalizing the mean fluorescence of the Pnt-GFP and Yan antibody channels by the mean fluorescence of the reference channel. This normalization serves to mitigate variability due to potentially uneven sample illumination, segment area, and in the case of His-RFP, differences in protein expression capacity between cells.

\textbf{Conversion of distance to time}

Cell positions along the anterior-posterior axis were mapped to developmental time as described previously (Peláez et al. 2015; Peláez 2016). This is predicated on two assumptions: the furrow proceeds at a constant rate of one column of R8 neurons per two hours, and minimal cell migration occurs. We find no reason to discard these assumptions.

For each disc, Delaunay triangulations were used to estimate the median distance between adjacent columns of R8 neurons (Fortune 1992). We used the median rather than the mean distance, as was used in our previous study, because it minimized the influence of non-adjacent R8s that were falsely identified by the triangulation. Dividing the furrow velocity of 2 h per column by this median distance yields a single conversion factor from position along the anterior-posterior axis to developmental time. This factor was applied to all cell measurements within the corresponding disc, yielding expression time series. Notably, these are not single cell dynamics, but rather aggregate dynamics across the development time course of a cell population.

\textbf{Computation of moving averages and confidence intervals}

Moving averages were computed by first-order Savitzky-Golay filtration (Savitzky and Golay 1964). This method augments the simple windowing approach used in our previous study (Peláez et al. 2015) by enabling visualization of expression trends at early time-points that are otherwise obscured by large window sizes. A secondary first-order filtration with one-fifth the original window size was applied to smooth lines for visualization purposes.

None of our conclusions are sensitive to our choice of filtration or smoothing method (Peláez et al. 2015). Primary window sizes of 250 and 75 cells were used for reporting the expression of multipotent and differentiated cells, unless noted otherwise. Confidence intervals for the moving average were inferred from the 2.5th and 97.5th percentile of 1000 point estimates of the mean within each window. Point estimates were generated by bootstrap resampling with replacement of the expression levels within each window.

\textbf{Alignment of expression data}

Cells were aligned with a reference population by shifting them in time. The magnitude of this shift was determined by maximizing the cross-correlation of progenitor Pnt-GFP expression \emph{Y(t)} with the corresponding reference time series \emph{X(t)}. Rather than raw measurements, moving averages within a window of ten cells were used to improve robustness against noise. This operation amounts to:

\includegraphics[width=1.44444in,height=0.25000in]{media/image1.png}

\includegraphics[width=3.29167in,height=0.36111in]{media/image2.png}

where, \emph{μ} and \emph{σ} are the mean and standard deviation of each time series, and \emph{dt} is the time shift by which the population should be shifted.

For each experimental treatment, a disc was randomly chosen and shifted in time such that time zero corresponds to the first annotated R8 neuron. This disc then served as the reference population for the alignment of all subsequent biological replicates within the treatment. Similarly, different experimental treatments (e.g. control and perturbation) were aligned by first aligning the discs within each treatment, then aggregating all cells within each treatment and repeating the procedure with the first treatment serving as the reference.

This approach differs from the previous implementation of our pipeline in which discs were manually aligned by the inflection point of their Yan-YFP expression profiles. Manual alignment entails arbitrary prioritization of certain dynamic features over others. Our revised protocol yields consistent, reproducible alignment of expression time series that equally weighs the entire time course. The automated approach is more principled but less robust than the manual approach. Specifically, it fails when dynamic forms qualitatively differ between experimental treatments. In these instances, we revert to manual alignment using the inflection point of Pnt-GFP induction as a reference.

\textbf{Simple competitive binding model}

Figure S4 presents results for an equilibrium model of two species, Yan (Y) and Pnt (P), competing for a finite pool of shared binding sites, S:

\[P + S\text{SP}\]

where K\textsubscript{D,Yan} and K\textsubscript{D,Pnt} are equilibrium association constants and SY and SP denote the bound species. Applying a mass balance to the total protein and binding site (S\textsubscript{0}) abundances:

\[Y_{0} = Y + SY\]

\[P_{0} = P + SP\]

\[S_{0} = S + SY + SP\]

yields an analytically tractable system of nonlinear equations (Wang 1995). For each pair of absolute protein abundances (Y\textsubscript{0}, P\textsubscript{0}) in Figure S4B, the Pnt binding site occupancy is simply SP/S\textsubscript{0}.

\textbf{Competitive binding model with cooperativity}

The model presented in Figure 3 expands upon the work of Hope, Rebay, and Reinitz (2017). The model is based on a single cis-regulatory element consisting of \emph{n} adjacent binding sites, each of which may be designated as ETS or non-ETS. Each binding site may only exist in one of three binding states; bound by a single copy of Yan, bound by a single copy of Pnt, or unbound. Thermodynamic potentials were assigned to each binding state using two parameters for each transcription factor. The parameter \emph{α\textsubscript{X}} defines the free energy of transcription factor \emph{X} binding to an ETS site, while \emph{β\textsubscript{X}} defines the free energy of binding to a non-ETS site (Fig. S5A). A unique configuration of binding states for all \emph{n} binding sites constitutes a single microstate, \emph{k}. The thermodynamic potential of each microstate was taken to be the sum of thermodynamic potentials for each of its constituent binding sites. For each microstate, the stabilizing effect of polymerization was incorporated via a third parameter, \emph{γ\textsubscript{X}}, that defines the free energy of SAM-SAM binding between a pair of similar transcription factors bound to adjacent sites. The net result is a total thermodynamic potential, \emph{ΔG\textsubscript{k}}, for each microstate. An example enumeration of all possible microstates for an element consisting of one ETS site preceding two non-ETS sites is provided in Figure S5B. The statistical frequencies of each microstate were evaluated by constructing a canonical ensemble:

\includegraphics[width=2.52778in,height=0.76389in]{media/image3.png}

in which \emph{p\textsubscript{k}} is the statistical frequency of microstate \emph{k}, \emph{{[}P{]}} is the Pnt concentration, \emph{{[}Y{]}} is the Yan concentration, \emph{a\textsubscript{P}(k)} and \emph{a\textsubscript{Y}(k)} are functions representing the number of bound molecules of \emph{P} and \emph{Y} within microstate k, \emph{T} is a fixed temperature set to 300 K, and \emph{R} is the gas constant. Fractional occupancies for each binding site correspond to the cumulative statistical frequency of all microstates in which the site is occupied by a given transcription factor. Overall fractional occupancies are similarly evaluated across all sites within the element.

We consider regulatory elements comprised of 12 binding sites in which only the first site carries the ETS designation. We retain the same parameterization of Yan binding proposed by Hope, Rebay, and Reinitz (2017): \emph{α\textsubscript{Y}}=-9.955 kcal mol\textsuperscript{-1}, \emph{β\textsubscript{Y}}=-5.837 kcal mol\textsuperscript{-1}, and \emph{γ\textsubscript{Y}}=-7.043 kcal mol\textsuperscript{-1}. We parameterized Pnt binding thermodynamics to provide balanced competition between Pnt and Yan in the absence of any SAM-mediated polymerization of Pnt. That is, we set Pnt binding affinities such that the transition from Pnt to Yan occupancy occurs when Pnt and Yan concentrations are approximately equal. While a parameterization using experimentally measured data would improve predictive accuracy, our aim here is primarily to obtain insight. The model used to generate Fig. 3D-F assumes that Pnt binds individual sites with elevated affinities \emph{α\textsubscript{P}} = 0.96(\emph{α\textsubscript{Y}} + \emph{γ\textsubscript{Y}}) and \emph{β\textsubscript{P}} = 0.96(\emph{β\textsubscript{Y}} + \emph{γ\textsubscript{Y}}). The model used to generate Fig. 3A-C uses these same elevated binding affinities for Yan, while setting \emph{γ\textsubscript{Y}}=0 kcal mol\textsuperscript{-1}. Qualitatively, our results are not sensitive to this parameterization.

\textbf{Analysis of \emph{yan} clones}

We used \emph{ey\textgreater{}FLP} and \emph{FRT40A} to generate \emph{yan\textsuperscript{833}} null clones within 23 eye discs carrying the Pnt-GFP transgene (see Genetics section). The chromosome carrying the wildtype \emph{yan} allele was marked with a Ubi-mRFPnls transgene, enabling automated detection of subpopulations with distinct \emph{yan} gene dosages, each characterized by a distinct level of mRFP fluorescence (Fig Sl-1A,B, Sl-2A). Discs were dissected, fixed, and co-stained with DAPI prior to confocal imaging. Images of 36 unique vertical cross-sections spanning non-overlapping cells were collected in total. Images were segmented using previously published open-source code (van der Walt et al. 2014). For each image, a foreground mask was constructed by Otsu thresholding the DAPI fluorescence channel following a series of smoothing and contrast-limited adaptive histogram equalization operations. Cell nuclei were identified by watershed segmentation of these foreground masks following a Euclidean distance-transform, with local maxima serving as seeds. Segments containing less than 250 pixels were removed. For each segment, Ubi-mRFPnls and Pnt-GFP fluorescence was quantified by normalizing the average intensity of all pixels within the respective fluorescence channel by the average DAPI fluorescence. A custom-made graphical user interface was used to manually draw boundaries around regions immediately posterior to the MF in each eye disc.

There was fluorescence bleed-through between the GFP and RFP channels (Fig. SI-2A). To correct for this, control clones were generated in six wildtype \emph{yan} eye discs co-expressing Pnt-GFP and Ubi-mRFPnls. After imaging and annotation, we implemented a procedure to systematically correct for bleed-through within each image (Fig. SI-2B). Our bleed-through correction strategy is predicated on the assumption that the fluorescence intensity \emph{F\textsubscript{i,j}} for each channel \emph{i} at each pixel \emph{j} is given by the sum of a background intensity \emph{B\textsubscript{i,j}} and some function of the expression level \emph{E\textsubscript{i,j}} that we wish to compare between cells (McMullen, Morimoto, and Amaral 2010):

\includegraphics[width=1.34722in,height=0.16667in]{media/image4.png}

Our bleed-through controls suggest that the background intensity for the Pnt-GFP channel \emph{B\textsubscript{GFP,j}} depends upon the fluorescence intensity of the Ubi-mRFPnls channel \emph{F\textsubscript{RFP,j}}. We assume this dependence is linear within a single image:

\includegraphics[width=1.54167in,height=0.16667in]{media/image5.png}

By using the background pixels to estimate the parameters \emph{α} and \emph{β} we may compare relative expression levels between nuclei by subtracting the background trend:

\includegraphics[width=3.01389in,height=0.18056in]{media/image6.png}

where 〈\ldots{}〉 denotes the average across all pixels \emph{j} comprising an individual cell measurement.

Foreground masks of each image were morphologically dilated until no features remained visible in the background (Fig. SI-2B). Background pixels were extracted and resampled such that the distribution of pixel RFP intensities was approximately uniform. Resampling was necessary due to the overwhelming abundance of zero-intensity pixels in the background. We then estimated the parameters \emph{α} and \emph{β} for each image by fitting a generalized linear model to the red and green fluorescence intensities of the resampled pixels. These models were formulated using identity link functions under the assumption that residuals were gamma distributed. Model fits were then used to correct the measured Pnt-GFP intensities within each cell. Corrected Pnt-GFP intensities were further normalized by DAPI to mitigate variation in gene expression capacity across cells.

Bleed-through correction successfully eliminated any detectable difference in Pnt-GFP expression between Ubi-mRFPnls genotypes in the wildtype \emph{yan} control discs (Fig. SI-2C). The same procedure was applied to all measurements of \emph{yan} null clones (Figure SI-2D).

Mitotic recombination between the mutant and wildtype \emph{yan} chromosomes yields cell subpopulations exhibiting low, medium, and high levels of Ubi-mRFPnls. These correspond to cells with 0,1, and 2 copies of the wildtype \emph{yan} allele, respectively. Segmented nuclei were assigned to one of three groups using a k-means classifier (Fig. SI-1A). In some images, cells naturally clustered about three modes of Ubi-mRFPnls fluorescence, and three seeds provided adequate separation of the clusters. Where necessary, additional seeds were manually added and the resultant clusters were merged in order to align the classification with visual inspection of the distribution of Ubi-mRFPnls levels. The procedure was validated through comparison with manual annotation of \textasciitilde{}2500 cells measured in four eye discs. The overall misclassification rate was \textasciitilde{}5\%, but errors were concentrated between the medium and high Ubi-mRFPnls types. We are consequently hesitant to compare Pnt-GFP levels between cells with one and two copies of the wildtype \emph{yan} allele.

Cells residing on the border of each clone were excluded from all comparisons to mitigate edge effects (Fig. SI-1B). Border cells were defined as those connected to a different genotype by an edge within a Delaunay triangulation of all cell positions in a particular image. Comparisons were further limited to cells taken from regions immediately posterior to the MF in which the compared clonal genotypes overlap in developmental time. This restriction served to buffer against differences in developmental context and focus attention on the region of elevated Pnt-GFP expression. The remaining protein measurements were aggregated across all eye discs for comparison between genotypes.

\textbf{Visualization of relative Pnt and Yan expression in \emph{Notch} mutant discs}

Visualizations were constructed by applying a smoothing operation to maximum intensity projections across confocal layers spanning progenitors (Fig. S5 A,B), and then mapping the absolute difference in Pnt-GFP and Yan antibody fluorescence to a diverging color scale. The smoothing operation consists of three sequential applications of a grey-closing filter followed by a single pass of a three-pixel wide median filter. This procedure dampens noise. Raw image fluorescence intensities were normalized to a 0-1 scale before application of any filters, so the maximum possible difference between Pnt-GFP and Yan fluorescence channels is unity. The color scale was truncated to a range of -0.3 to 0.3 for visualization purposes. No post-processing was applied to the maximum intensity projections in Figures S5 A and S5 B.

\textbf{Analysis of periodic spatial patterns in \emph{Notch} mutant discs}

Progenitor cells were selected from a 1.75 h window immediately posterior to the morphogenetic furrow. This window corresponds to approximately one column of eventual ommatidia. The window is identifiable in \emph{Notch} mutant discs because the MF serves as a reference. Digital spatial signals were assembled by sampling progenitor log\textsubscript{2}-transformed Pnt to Yan ratios, \emph{X}, as a function of cell position along the dorso-ventral axis, \emph{y}. Both autocorrelation analysis and spectral decomposition were applied to these signals.

Autocorrelation functions were assembled by computing the moving average of expression similarity, \emph{C}, as a function of dorso-ventral separation distance, \emph{d}:

\includegraphics[width=2.13889in,height=0.37500in]{media/image7.emf}

\includegraphics[width=0.94444in,height=0.16667in]{media/image8.emf}

where \emph{E} denotes the expected value, and \emph{i} and \emph{j} are indexed over all cells in order of increasing separation distance. Moving averages and confidence intervals were computed as described previously, with a window size of 50 sequential values.

Spatial signals were decomposed into spectral components via the Lomb-Scargle periodogram using the AstroML software package (VanderPlas et al. 2012). These periodograms were used rather than Fourier decomposition because they enable spectral decomposition of irregularly sampled signals (VanderPlas 2018). Significance thresholds were inferred from the 95th percentile of peak spectral powers detected during repeated decomposition of 1000 null signals. Null signals were constructed by resampling signal intensities while maintaining constant sampling times.

\textbf{Data and software availability}

All segmented and annotated eye discs will be made available via our data repository upon publication. The \emph{FlyEye Silhouette} software will be made freely available via the macOS App Store. All subsequent expression analysis and visualization procedures, our transcription factor binding model, and clones analysis pipeline will be made available as open-source Python packages. We will also provide a repository that includes a series of Jupyter notebooks that enable complete reproduction of all figures and analysis presented in the manuscript.

\textbf{Acknowledgements:}

We thank Kevin White's lab and the MODEncode Consortium for recombineering the Pnt-GFP transgene, the Howard Hughes Medical Institute (N.P.), the Chicago Biomedical Consortium (N.P), the Robert Lurie Northwestern Cancer Center (N.P.), and the NIH (EY025957, I.R. and R.W.C; GM118144, R.W.C.) for funding, Benny Shilo (Weizmann Institute) for providing stocks carrying the PntP1 (HS20) reporter, the Bloomington Drosophila Stock Center for the remaining stocks, Laura Nilson for the use of computational resources, the Developmental Studies Hybridoma Bank for antibody reagents, and the Northwestern Biological Imaging Facility (BIF) for technical imaging support.

\textbf{REFERENCES}

Adler, Miri, and Uri Alon. 2018. ``Fold-Change Detection in Biological Systems.'' \emph{Current Opinion in Systems Biology} 8 (April). Elsevier: 81--89. doi:10.1016/j.coisb.2017.12.005.

Antebi, Yaron E., James M. Linton, Heidi Klumpe, Bogdan Bintu, Mengsha Gong, Christina Su, Reed McCardell, and Michael B. Elowitz. 2017. ``Combinatorial Signal Perception in the BMP Pathway.'' \emph{Cell} 170 (September). Cell Press: 1184--96. doi:10.1016/j.cell.2017.08.015.

Baker, Nicholas E., and Sung-Yun Yu. 1997. ``Proneural Function of Neurogenic Genes in the Developing Drosophila Eye.'' \emph{Current Biology} 7 (February). Cell Press: 122--32. doi:10.1016/S0960-9822(06)00056-X.

Basler, Konrad, and Ernst Hafen. 1989. ``Dynamics of Drosophila Eye Development and Temporal Requirements of Sevenless Expression.'' \emph{Development} 107: 723--31.

Boisclair Lachance, Jean-François, Nicolás Peláez, Justin J. Cassidy, Jemma L. Webber, Ilaria Rebay, and Richard W. Carthew. 2014. ``A Comparative Study of Pointed and Yan Expression Reveals New Complexity to the Transcriptional Networks Downstream of Receptor Tyrosine Kinase Signaling.'' \emph{Developmental Biology} 385 (January): 263--78. doi:10.1016/j.ydbio.2013.11.002.

Boisclair Lachance, Jean-François, Jemma L Webber, Lu Hong, Aaron R Dinner, and Ilaria Rebay. 2018. ``Cooperative Recruitment of Yan via a High-Affinity ETS Supersite Organizes Repression to Confer Specificity and Robustness to Cardiac Cell Fate Specification.'' \emph{Genes and Development} 32 (March). Cold Spring Harbor Laboratory Press: 389--401. doi:10.1101/gad.307132.117.

Brunner, D, K Dücker, N Oellers, E Hafen, H Scholz, and C Klämbt. 1994. ``The ETS Domain Protein Pointed-P2 Is a Target of MAP Kinase in the Sevenless Signal Transduction Pathway.'' \emph{Nature} 370 (6488): 386--89. doi:10.1038/370386a0.

Bush, Alan, Gustavo Vasen, Andreas Constantinou, Paula Dunayevich, Inés Lucía Patop, Matías Blaustein, and Alejandro Colman‐Lerner. 2016. ``Yeast GPCR Signaling Reflects the Fraction of Occupied Receptors, Not the Number.'' \emph{Molecular Systems Biology} 12 (December). EMBO Press: 898. doi:10.15252/msb.20166910.

Cagan, Ross L., and Donald F. Ready. 1989. ``Notch Is Required for Successive Cell Decisions in the Developing Drosophila Retina.'' \emph{Genes \& Development} 3: 1099--1112. doi:10.1101/gad.3.8.1099.

Campos-Ortega, J. A., and A. Hofbauer. 1977. ``Cell Clones and Pattern Formation: On the Lineage of Photoreceptor Cells in the Compound Eye of Drosophila.'' \emph{Roux's Archives of Developmental Biology} 181: 227--45.

Clifford, R.J., and T. Schupbach. 1989. ``Coordinately and Differentially Mutable Activities of Torpedo, the Drosophila Melanogaster Homolog of the Vertebrate EGF Receptor Gene.'' \emph{Genetics} 123 (4): 771--87. doi:10.1016/0168-9525(90)90089-O.

Doe, Chris Q. 2017. ``Temporal Patterning in the Drosophila CNS.'' \emph{Annual Review of Cell and Developmental Biology} 33 (1). Annual Reviews: 219--40. doi:10.1146/annurev-cellbio-111315-125210.

Ducy, Patricia, Rui Zhang, Valérie Geoffroy, Amy L. Ridall, and Gérard Karsenty. 1997. ``Osf2/Cbfa1: A Transcriptional Activator of Osteoblast Differentiation.'' \emph{Cell} 89 (5). Cell Press: 747--54. doi:10.1016/S0092-8674(00)80257-3.

Erclik, Ted, Xin Li, Maximilien Courgeon, Claire Bertet, Zhenqing Chen, Ryan Baumert, June Ng, et al. 2017. ``Integration of Temporal and Spatial Patterning Generates Neural Diversity.'' \emph{Nature} 541 (January). Nature Publishing Group: 365--70. doi:10.1038/nature20794.

Evans, Rachel A., Y. Chung Tian, Robert Steadman, and Aled O. Phillips. 2003. ``TGF-Β1-Mediated Fibroblast-Myofibroblast Terminal Differentiation - The Role of Smad Proteins.'' \emph{Experimental Cell Research} 282 (2). Academic Press: 90--100. doi:10.1016/S0014-4827(02)00015-0.

Flores, Gail V, Hong Duan, Huajun Yan, Raghavendra Nagaraj, Weimin Fu, Yu Zou, Markus Noll, and Utpal Banerjee. 2000. ``Combinatorial Signaling in the Specification of Unique Cell Fates.'' \emph{Cell} 103 (September). Cell Press: 75--85. doi:10.1016/S0092-8674(00)00106-9.

Fortini, Mark E., Ilaria Rebay, Laurent A. Caron, and Spyros Artavanis-Tsakonas. 1993. ``An Activated Notch Receptor Blocks Cell-Fate Commitment in the Developing Drosophila Eye.'' \emph{Nature} 365 (October). Nature Publishing Group: 555--57. doi:10.1038/365555a0.

Fortini, Mark E., Michael A. Simon, and Gerald M. Rubin. 1992. ``Signalling by the Sevenless Protein Tyrosine Kinase Is Mimicked by Ras1 Activation.'' \emph{Nature} 355: 559--61. doi:10.1038/355559a0.

Fortune, Steven. 1992. ``Voronoi Diagrams and Delaunay Triangulations.'' \emph{Computing in Euclidean Geometry}. doi:10.1016/0098-3004(93)90062-A.

Freeman, Matthew. 1996. ``Reiterative Use of the EGF Receptor Triggers Differentiation of All Cell Types in the Drosophila Eye.'' \emph{Cell} 87 (November). Cell Press: 651--60. doi:10.1016/S0092-8674(00)81385-9.

Frick, Christopher L., Clare Yarka, Harry Nunns, and Lea Goentoro. 2017. ``Sensing Relative Signal in the Tgf-β/Smad Pathway.'' \emph{Proceedings of the National Academy of Sciences} 114 (14). National Academy of Sciences: E2975--82. doi:10.1073/pnas.1611428114.

Gabay, Limor, Henrike Scholz, Myriam Golembo, Andrea Klaes, Ben-Zion Shilo, and Christian Klambt. 1996. ``EGF Receptor Signaling Induces Pointed P1 Transcription and Inactivates Yan Protein in the Drosophila Embryonic Ventral Ectoderm.'' \emph{Development} 122: 3355--62.

Gavish, Avishai, Arkadi Shwartz, Abraham Weizman, Eyal Schejter, Ben-Zion Shilo, and Naama Barkai. 2016. ``Periodic Patterning of the Drosophila Eye Is Stabilized by the Diffusible Activator Scabrous.'' \emph{Nature Communications} 7:10461. doi:10.1038/ncomms10461.

Goentoro, Lea, and Marc W. Kirschner. 2009. ``Evidence That Fold-Change, and Not Absolute Level, of β-Catenin Dictates Wnt Signaling.'' \emph{Molecular Cell} 36 (5). Cell Press: 872--84. doi:10.1016/j.molcel.2009.11.017.

Goentoro, Lea, Oren Shoval, Marc W. Kirschner, and Uri Alon. 2009. ``The Incoherent Feedforward Loop Can Provide Fold-Change Detection in Gene Regulation.'' \emph{Molecular Cell} 36 (December). Cell Press: 894--99. doi:10.1016/j.molcel.2009.11.018.

Graham, Thomas G. W., S. M. Ali Tabei, Aaron R. Dinner, and Ilaria Rebay. 2010. ``Modeling Bistable Cell-Fate Choices in the Drosophila Eye: Qualitative and Quantitative Perspectives.'' \emph{Development} 137 (14): 2265--78. doi:10.1242/dev.044826.

Halfon, Marc S., Ana Carmena, Stephen Gisselbrecht, Charles M. Sackerson, Fernando Jiménez, Mary K. Baylies, and Alan M. Michelson. 2000. ``Ras Pathway Specificity Is Determined by the Integration of Multiple Signal-Activated and Tissue-Restricted Transcription Factors.'' \emph{Cell} 103 (1). Cell Press: 63--74. doi:10.1016/S0092-8674(00)00105-7.

Hope, Matthew C., Ilaria Rebay, and John Reinitz. 2017. ``DNA Occupancy of Polymerizing Transcription Factors: A Chemical Model of the ETS Family Factor Yan.'' \emph{Biophysical Journal} 112 (1). Cell Press: 180--92. doi:10.1016/j.bpj.2016.11.901.

Jarman, Andrew P., Ellsworth H. Grell, Larry Ackerman, Lily Y. Jan, and Yuh Nung Jan. 1994. ``Atonal Is the Proneural Gene for Drosophila Photoreceptors.'' \emph{Nature} 369 (6479): 398--400. http://www.nature.com/nature/journal/v369/n6479/pdf/369398a0.pdf.

Kamino, Keita, Yohei Kondo, Akihiko Nakajima, Mai Honda-Kitahara, Kunihiko Kaneko, and Satoshi Sawai. 2017. ``Fold-Change Detection and Scale Invariance of Cell--Cell Signaling in Social Amoeba.'' \emph{Proceedings of the National Academy of Sciences} 114 (21). National Academy of Sciences: E4149--57. doi:10.1073/pnas.1702181114.

Kim, Chongwoo A., Martin L. Phillips, Woojae Kim, Mari Gingery, Hoang H. Tran, Michael A. Robinson, Salem Faham, and James U. Bowie. 2001. ``Polymerization of the SAM Domain of TEL in Leukemogenesis and Transcriptional Repression.'' \emph{EMBO Journal} 20 (15): 4173--82. doi:10.1093/emboj/20.15.4173.

Klämbt, C. 1993. ``The Drosophila Gene Pointed Encodes Two ETS-like Proteins Which Are Involved in the Development of the Midline Glial Cells.'' \emph{Development} 117 (1): 163--76.

Kueh, Hao Y., Ameya Champhekar, Stephen L. Nutt, Michael B. Elowitz, and Ellen V. Rothenberg. 2013. ``Positive Feedback between PU.1 and the Cell Cycle Controls Myeloid Differentiation.'' \emph{Science} 341 (August): 670--73. doi:10.1126/science.1240831.

Kumar, Justin P., Frank Hsiung, Maureen A. Powers, and Kevin Moses. 2003. ``Nuclear Translocation of Activated MAP Kinase Is Developmentally Regulated in the Developing Drosophila Eye.'' \emph{Development (Cambridge, England)} 130 (16). The Company of Biologists Ltd: 3703--14. doi:10.1242/dev.00556.

Kumar, Justin P., Murni Tio, Frank Hsiung, Sevak Akopyan, Limor Gabay, Rony Seger, Ben-Zion Shilo, and Kevin Moses. 1998. ``Dissecting the Roles of the Drosophila EGF Receptor in Eye Development and MAP Kinase Activation.'' \emph{Development} 125 (19): 3875--85.

Laslo, Peter, Chauncey J. Spooner, Aryeh Warmflash, David W. Lancki, Hyun-Jun Lee, Roger Sciammas, Benjamin N. Gantner, Aaron R. Dinner, and Harinder Singh. 2006. ``Multilineage Transcriptional Priming and Determination of Alternate Hematopoietic Cell Fates.'' \emph{Cell} 126 (4): 755--66. doi:10.1016/j.cell.2006.06.052.

Li, Yanxia, and Nicholas E. Baker. 2001. ``Proneural Enhancement by Notch Overcomes Suppressor-of-Hairless Repressor Function in the Developing Drosophila Eye.'' \emph{Current Biology} 11 (5). Cell Press: 330--38. doi:10.1016/S0960-9822(01)00093-8.

Lin, Yihan, Chang Ho Sohn, Chiraj K. Dalal, Long Cai, and Michael B. Elowitz. 2015. ``Combinatorial Gene Regulation by Modulation of Relative Pulse Timing.'' \emph{Nature} 527 (November). Nature Publishing Group: 54--58. doi:10.1038/nature15710.

Lubensky, David K., Matthew W. Pennington, Boris I. Shraiman, and Nicholas E. Baker. 2011. ``A Dynamical Model of Ommatidial Crystal Formation.'' \emph{Proceedings of the National Academy of Sciences} 108 (27): 11145--50. doi:10.1073/pnas.1015302108.

Lucchesi, John C., and John M. Rawls. 1973. ``Regulation of Gene Function: A Comparison of Enzyme Activity Levels in Relation to Gene Dosage in Diploids and Triploids of Drosophila Melanogaster.'' \emph{Biochemical Genetics} 9 (1): 41--51. doi:10.1007/BF00485589.

McGhee, James D., Tetsunari Fukushige, Michael W. Krause, Stephanie E. Minnema, Barbara Goszczynski, Jeb Gaudet, Yuji Kohara, et al. 2009. ``ELT-2 Is the Predominant Transcription Factor Controlling Differentiation and Function of the C. Elegans Intestine, from Embryo to Adult.'' \emph{Developmental Biology} 327 (2): 551--65. doi:10.1016/j.ydbio.2008.11.034.

McMullen, Patrick D., Richard I. Morimoto, and Luís A. Nunes Amaral. 2010. ``Physically Grounded Approach for Estimating Gene Expression from Microarray Data.'' \emph{Proceedings of the National Academy of Sciences} 107 (31). National Academy of Sciences: 13690--95. doi:10.1073/pnas.1000938107.

Melen, Gustavo J., Sagi Levy, Naama Barkai, and Ben-Zion Shilo. 2005. ``Threshold Responses to Morphogen Gradients by Zero-Order Ultrasensitivity.'' \emph{Molecular Systems Biology}, 1--11. doi:10.1038/msb4100036.

Morimoto, Alyssa M., Katherine C. Jordan, Kyria Tietze, Jessica S. Britton, Elizabeth M. O'Neill, and Hannele Ruohola-Baker. 1996. ``Pointed, an ETS Domain Transcription Factor, Negatively Regulates the EGF Receptor Pathway in Drosophila Oogenesis.'' \emph{Development} 122: 3745--54.

Nagaraj, Raghavendra, Jude Canon, and Utpal Banerjee. 2002. ``Cell Fate Specification in the Drosophila Eye.'' In \emph{Results and Problems in Cell Differentiation, Vol. 37}, 73--88. Springer, Berlin, Heidelberg. doi:10.1007/978-3-540-45398-7\_6.

Nitta, Kazuhiro R., Arttu Jolma, Yimeng Yin, Ekaterina Morgunova, Teemu Kivioja, Junaid Akhtar, Korneel Hens, et al. 2015. ``Conservation of Transcription Factor Binding Specificities across 600 Million Years of Bilateria Evolution.'' \emph{ELife} 4. doi:10.7554/eLife.04837.

Niwa, Hitoshi, Jun-Ichi Miyazaki, and Austin G. Smith. 2000. ``Quantitative Expression of Oct-3/4 Defines Differentiation, Dedifferentiation or Self-Renewal of ES Cells.'' \emph{Nature Genetics} 24 (April): 372--76. https://www.nature.com/articles/ng0400\_372.pdf.

O'Neill, Elizabeth M., Ilaria Rebay, Robert Tjian, and Gerald Rubin. 1994. ``The Activities of Two ETS-Related Transcription Factors Required for Drosophila Eye Development Are Modulated by the Ras/MAPK Pathway.'' \emph{Cell} 78: 137--47.

O'Riordan, Mary, and Rudolf Grosschedl. 1999. ``Coordinate Regulation of B Cell Differentiation by the Transcription Factors EBF and E2A.'' \emph{Immunity} 11: 21--31. doi:10.1016/S1074-7613(00)80078-3.

Park, Byung Ouk, Robert Ahrends, and Mary N. Teruel. 2012. ``Consecutive Positive Feedback Loops Create a Bistable Switch That Controls Preadipocyte-to-Adipocyte Conversion.'' \emph{Cell Reports} 2 (4): 976--90. doi:10.1016/j.celrep.2012.08.038.

Peláez, Nicolás. 2016. ``Dynamics and Heterogeneity of Fate Determinants during Transition towards Cell Differentiation.'' Northwestern University.

Peláez, Nicolás, Arnau Gavalda-Miralles, Bao Wang, Heliodoro Tejedor Navarro, Herman Gudjonson, Ilaria Rebay, Aaron R. Dinner, Aggelos K. Katsaggelos, Luís A. Nunes Amaral, and Richard W. Carthew. 2015. ``Dynamics and Heterogeneity of a Fate Determinant during Transition towards Cell Differentiation.'' \emph{ELife} 4 (November). eLife Sciences Publications Limited. doi:10.7554/eLife.08924.

Qiao, Feng, Haiyun Song, Chongwoo A. Kim, Michael R. Sawaya, Jacob B. Hunter, Mari Gingery, Ilaria Rebay, Albert J. Courey, and James U. Bowie. 2004. ``Derepression by Depolymerization: Structural Insights into the Regulation of Yan by Mae.'' \emph{Cell} 118 (July). Cell Press: 163--73. doi:10.1016/j.cell.2004.07.010.

Raj, Arjun, Scott A. Rifkin, Erik Andersen, and Alexander van Oudenaarden. 2010. ``Variability in Gene Expression Underlies Incomplete Penetrance.'' \emph{Nature} 463 (7283): 913--18. doi:10.1038/nature08781.

Ready, Donald F., Thomas E. Hanson, and Seymour Benzer. 1976. ``Development of the Drosophila Retina, a Neurocrystalline Lattice.'' \emph{Developmental Biology} 53 (2): 217--40. doi:10.1016/0012-1606(76)90225-6.

Rebay, Ilaria, and Gerald M. Rubin. 1995. ``Yan Functions as a General Inhibitor of Differentiation and Is Negatively Regulated by Activation of the Ras1/MAPK Pathway.'' \emph{Cell} 81: 857--66.

Rogge, R., P. J. Green, J. Urano, S. Horn-Saban, M. Mlodzik, B Z Shilo, V. Hartenstein, and U. Banerjee. 1995. ``The Role of Yan in Mediating the Choice between Cell Division and Differentiation.'' \emph{Development} 121 (12): 3947--58.

Rohrbaugh, Margaret, Edward Ramos, Duc Nguyen, Mitch Price, Yu Wen, and Zhi-Chun Lai. 2002. ``Notch Activation of Yan Expression Is Antagonized by RTK/Pointed Signaling in the Drosophila Eye.'' \emph{Current Biology} 12: 576--81.

Savitzky, Abraham, and Marcel J.E. Golay. 1964. ``Smoothing and Differentiation of Data by Simplified Least Squares Procedures.'' \emph{Analytical Chemistry} 36 (8): 1627--39. doi:10.1021/ac60214a047.

Scholz, H., J Deatrick, A Klaes, and C Klämbt. 1993. ``Genetic Disection of Pointed, a Drosophila Gene Encoding Two ETS-Related Proteins.'' \emph{Genetics} 135 (2): 455--68.

Shellenbarger, David L., and Dawson J. Mohler. 1975. ``Temperature-Sensitive Mutations of the Notch Locus in Drosophila Melanogaster.'' \emph{Genetics} 81 (1). Genetics Society of America: 143--62.

Shwartz, Arkadi, Shaul Yogev, Eyal D. Schejter, and Ben-Zion Shilo. 2013. ``Sequential Activation of ETS Proteins Provides a Sustained Transcriptional Response to EGFR Signaling.'' \emph{Development} 140 (13): 2746--54. doi:10.1242/dev.093138.

Simon, Michael A., David D.L. Bowtell, G. Steven Dodson, Todd R. Laverty, and Gerald M. Rubin. 1991. ``Ras1 and a Putative Guanine Nucleotide Exchange Factor Perform Crucial Steps in Signaling by the Sevenless Protein Tyrosine Kinase.'' \emph{Cell} 67 (4). Cell Press: 701--16. doi:10.1016/0092-8674(91)90065-7.

Spitz, François, and Eileen E. M. Furlong. 2012. ``Transcription Factors: From Enhancer Binding to Developmental Control.'' \emph{Nature Reviews Genetics} 13 (9). Nature Publishing Group: 613--26. doi:10.1038/nrg3207.

Tomlinson, Andrew. 1985. ``The Cellular Dynamics of Pattern Formation in the Eye of Drosophila.'' \emph{Development} 89: 313--31. http://www.ncbi.nlm.nih.gov/pubmed/3937883.

Tomlinson, Andrew, and Donald F. Ready. 1987. ``Neuronal Differentiation in the Drosophila Ommatidium.'' \emph{Developmental Biology} 120 (2): 366--76. doi:10.1016/0012-1606(87)90239-9.

Tontonoz, Peter, Erding Hu, and Bruce M Spiegelman. 1994. ``Stimulation of Adipogenesis in Fibroblasts by PPARγ2, a Lipid-Activated Transcription Factor.'' \emph{Cell} 79 (7): 1147--56. doi:10.1016/0092-8674(94)90006-X.

VanderPlas, Jacob. 2018. ``Understanding the Lomb-Scargle Periodogram.'' \emph{The Astrophysical Journal Supplement Series} 236 (1): 1--55. http://arxiv.org/abs/1703.09824.

VanderPlas, Jacob, Andrew J. Connolly, Zeljko Ivezic, and Alex Gray. 2012. ``Introduction to AstroML: Machine Learning for Astrophysics.'' In \emph{Proceedings - CIDU 2012}, 47--54. doi:10.1109/CIDU.2012.6382200.

Voas, Matthew G., and Ilaria Rebay. 2004. ``Signal Integration during Development: Insights from the Drosophila Eye.'' \emph{Developmental Dynamics}. Wiley-Blackwell. doi:10.1002/dvdy.10449.

Walt, Stéfan van der, Johannes L. Schönberger, Juan Nunez-Iglesias, François Boulogne, Joshua D. Warner, Neil Yager, Emmanuelle Gouillart, and Tony Yu. 2014. ``Scikit-Image: Image Processing in Python.'' \emph{PeerJ}. doi:10.7717/peerj.453.

Wang, Zhi-Xin. 1995. ``An Exact Mathematical Expression for Describing Competitive Binding of Two Different Ligands to a Protein Molecule.'' \emph{FEBS Letters} 360: 111--14. doi:10.1016/0014-5793(95)00062-E.

Webber, Jemma L., Jie Zhang, Lauren Cote, Pavithra Vivekanand, Xiaochun Ni, Jie Zhou, Nicolas Negre, Richard W. Carthew, Kevin P. White, and Ilaria Rebay. 2013. ``The Relationship between Long-Range Chromatin Occupancy and Polymerization of the Drosophila ETS Family Transcriptional Repressor Yan.'' \emph{Genetics} 193 (2): 633--49. doi:10.1534/genetics.112.146647.

Webber, Jemma L., Jie Zhang, Aaron Mitchell-Dick, and Ilaria Rebay. 2013. ``3D Chromatin Interactions Organize Yan Chromatin Occupancy and Repression at the Even-Skipped Locus.'' \emph{Genes \& Development} 27 (21). Cold Spring Harbor Laboratory Press: 2293--98. doi:10.1101/gad.225789.113.

Webber, Jemma L, Jie Zhang, Alex Massey, Nicelio Sanchez-Luege, and Ilaria Rebay. 2018. ``Collaborative Repressive Action of the Antagonistic ETS Transcription Factors Pointed and Yan Fine-Tunes Gene Expression to Confer Robustness in Drosophila.'' \emph{Development} 145 (13). Oxford University Press for The Company of Biologists Limited: dev165985. doi:10.1242/dev.165985.

Wei, Gong Hong, Gwenael Badis, Michael F. Berger, Teemu Kivioja, Kimmo Palin, Martin Enge, Martin Bonke, et al. 2010. ``Genome-Wide Analysis of ETS-Family DNA-Binding in Vitro and in Vivo.'' \emph{EMBO Journal} 29 (13). EMBO Press: 2147--60. doi:10.1038/emboj.2010.106.

Wolff, Samuel C., Kasia M. Kedziora, Raluca Dumitru, Cierra D. Dungee, Tarek M. Zikry, Adriana S. Beltran, Rachel A. Haggerty, JrGang Cheng, Margaret A. Redick, and Jeremy E. Purvis. 2018. ``Inheritance of OCT4 Predetermines Fate Choice in Human Embryonic Stem Cells.'' \emph{Molecular Systems Biology} 14 (9). Cold Spring Harbor Laboratory: e8140. doi:10.1101/137299.

Wolff, T., and Donalf F. Ready. 1991. ``Cell Death in Normal and Rough Eye Mutants of Drosophila.'' \emph{Development} 113 (3): 825--39. doi:8223268.

Wolff, Tanya, and Donald F. Ready. 1991. ``The Beginning of Pattern Formation in the Drosophila Compound Eye: The Morphogenetic Furrow and the Second Mitotic Wave.'' \emph{Development} 113 (3): 841--50. http://www.ncbi.nlm.nih.gov/pubmed/1726564.

---------. 1993. \emph{Pattern Formation in the Drosophila Retina}. Edited by M. Bate and A. Martinez-Arias. \emph{In the Development of Drosophila Melanogaster Vol. 2}. Plainview, NY: Cold Spring Harbor Press.

Xu, Chunyan, Rachele C. Kauffmann, Jianjun Zhang, Susan Kladny, and Richard W. Carthew. 2000. ``Overlapping Activators and Repressors Delimit Transcriptional Response to Receptor Tyrosine Kinase Signals in the Drosophila Eye.'' \emph{Cell} 103: 87--97. doi:10.1016/S0092-8674(00)00107-0.

Yang, Lihui, and Nicholas E. Baker. 2006. ``Notch Activity Opposes Ras-Induced Differentiation during the Second Mitotic Wave of the Developing Drosophila Eye.'' \emph{BMC Developmental Biology} 6 (8). BioMed Central: 1--10. doi:10.1186/1471-213X-6-8.

Yao, Guang, Tae Jun Lee, Seiichi Mori, Joseph R. Nevins, and Lingchong You. 2008. ``A Bistable Rb-E2F Switch Underlies the Restriction Point.'' \emph{Nature Cell Biology} 10 (4): 476--82. doi:10.1038/ncb1711.

Zhang, Jie, Thomas G.W. Graham, Pavithra Vivekanand, Lauren Cote, Maureen Cetera, and Ilaria Rebay. 2010. ``Sterile Alpha Motif Domain-Mediated Self-Association Plays an Essential Role in Modulating the Activity of the Drosophila ETS Family Transcriptional Repressor Yan.'' \emph{Molecular and Cellular Biology} 30 (5). American Society for Microbiology: 1158--70. doi:10.1128/MCB.01225-09.

Zhang, Tianyi, Swati Ranade, Chuan Qi Cai, Christopher Clouser, and Francesca Pignoni. 2006. ``Direct Control of Neurogenesis by Selector Factors in the Fly Eye: Regulation of Atonal by Ey and So.'' \emph{Development} 133 (24). Oxford University Press for The Company of Biologists Limited: 4881--89. doi:10.1242/dev.02669.

Zheng, Wei-ping, and Richard A. Flavell. 1997. ``The Transcription Factor GATA-3 Is Necessary and Sufficient for Th2 Cytokine Gene Expression in CD4 T Cells.'' \emph{Cell} 89 (4): 587--96. doi:10.1016/S0092-8674(00)80240-8.
